// intl-tel-input/svelte/build/IntlTelInput.mjs
var Xn = Object.defineProperty;
var ke = (e) => {
  throw TypeError(e);
};
var Qn = (e, t, n) => t in e ? Xn(e, t, { enumerable: !0, configurable: !0, writable: !0, value: n }) : e[t] = n;
var et = (e, t, n) => Qn(e, typeof t != "symbol" ? t + "" : t, n), re = (e, t, n) => t.has(e) || ke("Cannot " + n);
var T = (e, t, n) => (re(e, t, "read from private field"), n ? n.call(e) : t.get(e)), X = (e, t, n) => t.has(e) ? ke("Cannot add the same private member more than once") : t instanceof WeakSet ? t.add(e) : t.set(e, n), nt = (e, t, n, i) => (re(e, t, "write to private field"), i ? i.call(e, n) : t.set(e, n), n), wt = (e, t, n) => (re(e, t, "access private method"), n);
import { onMount as Zn, onDestroy as Jn } from "svelte";
const ti = "5";
var rn;
typeof window < "u" && ((rn = window.__svelte ?? (window.__svelte = {})).v ?? (rn.v = /* @__PURE__ */ new Set())).add(ti);
const ei = 1, ni = 4, ii = 8, si = 16, ri = 2, P = Symbol(), oi = "http://www.w3.org/1999/xhtml", li = "@attach", fe = !1;
var on = Array.isArray, ai = Array.prototype.indexOf, ui = Object.defineProperty, qt = Object.getOwnPropertyDescriptor, ci = Object.getOwnPropertyDescriptors, di = Object.prototype, fi = Array.prototype, ln = Object.getPrototypeOf;
function hi(e) {
  for (var t = 0; t < e.length; t++)
    e[t]();
}
function an() {
  var e, t, n = new Promise((i, s) => {
    e = i, t = s;
  });
  return { promise: n, resolve: e, reject: t };
}
const k = 2, Qt = 4, be = 8, Ie = 1 << 24, gt = 16, mt = 32, Rt = 64, un = 128, Y = 512, U = 1024, j = 2048, at = 4096, Nt = 8192, tt = 16384, Ee = 32768, Me = 65536, Ue = 1 << 17, cn = 1 << 18, se = 1 << 19, pi = 1 << 20, Ct = 32768, he = 1 << 21, Ne = 1 << 22, st = 1 << 23, ht = Symbol("$state"), _i = Symbol("legacy props"), Ci = Symbol(""), bt = new class extends Error {
  constructor() {
    super(...arguments);
    et(this, "name", "StaleReactionError");
    et(this, "message", "The reaction that called `getAbortSignal()` was re-run or destroyed");
  }
}();
function yi() {
  throw new Error("https://svelte.dev/e/async_derived_orphan");
}
function gi(e) {
  throw new Error("https://svelte.dev/e/effect_in_teardown");
}
function mi() {
  throw new Error("https://svelte.dev/e/effect_in_unowned_derived");
}
function vi(e) {
  throw new Error("https://svelte.dev/e/effect_orphan");
}
function wi() {
  throw new Error("https://svelte.dev/e/effect_update_depth_exceeded");
}
function bi(e) {
  throw new Error("https://svelte.dev/e/props_invalid_value");
}
function Ii() {
  throw new Error("https://svelte.dev/e/state_descriptors_fixed");
}
function Ei() {
  throw new Error("https://svelte.dev/e/state_prototype_fixed");
}
function Ni() {
  throw new Error("https://svelte.dev/e/state_unsafe_mutation");
}
function Di() {
  console.warn("https://svelte.dev/e/select_multiple_invalid_value");
}
function dn(e) {
  return e === this.v;
}
function Ai(e, t) {
  return e != e ? t == t : e !== t || e !== null && typeof e == "object" || typeof e == "function";
}
function Li(e) {
  return !Ai(e, this.v);
}
let Ti = !1, Q = null;
function Zt(e) {
  Q = e;
}
function Si(e, t = !1, n) {
  Q = {
    p: Q,
    i: !1,
    c: null,
    e: null,
    s: e,
    x: null,
    l: null
  };
}
function Pi(e) {
  var t = (
    /** @type {ComponentContext} */
    Q
  ), n = t.e;
  if (n !== null) {
    t.e = null;
    for (var i of n)
      En(i);
  }
  return e !== void 0 && (t.x = e), t.i = !0, Q = t.p, e ?? /** @type {T} */
  {};
}
function fn() {
  return !0;
}
let It = [];
function Oi() {
  var e = It;
  It = [], hi(e);
}
function Bt(e) {
  if (It.length === 0) {
    var t = It;
    queueMicrotask(() => {
      t === It && Oi();
    });
  }
  It.push(e);
}
function Ri(e) {
  var t = D;
  if (t === null)
    return w.f |= st, e;
  if ((t.f & Ee) === 0) {
    if ((t.f & un) === 0)
      throw e;
    t.b.error(e);
  } else
    Jt(e, t);
}
function Jt(e, t) {
  for (; t !== null; ) {
    if ((t.f & un) !== 0)
      try {
        t.b.error(e);
        return;
      } catch (n) {
        e = n;
      }
    t = t.parent;
  }
  throw e;
}
const ki = -7169;
function x(e, t) {
  e.f = e.f & ki | t;
}
function De(e) {
  (e.f & Y) !== 0 || e.deps === null ? x(e, U) : x(e, at);
}
function hn(e) {
  if (e !== null)
    for (const t of e)
      (t.f & k) === 0 || (t.f & Ct) === 0 || (t.f ^= Ct, hn(
        /** @type {Derived} */
        t.deps
      ));
}
function Mi(e, t, n) {
  (e.f & j) !== 0 ? t.add(e) : (e.f & at) !== 0 && n.add(e), hn(e.deps), x(e, U);
}
const $t = /* @__PURE__ */ new Set();
let S = null, R = null, z = [], Ae = null, pe = !1;
var Dt, At, ft, Lt, jt, Tt, St, Pt, J, _e, Ce, pn;
const Re = class Re {
  constructor() {
    X(this, J);
    et(this, "committed", !1);
    /**
     * The current values of any sources that are updated in this batch
     * They keys of this map are identical to `this.#previous`
     * @type {Map<Source, any>}
     */
    et(this, "current", /* @__PURE__ */ new Map());
    /**
     * The values of any sources that are updated in this batch _before_ those updates took place.
     * They keys of this map are identical to `this.#current`
     * @type {Map<Source, any>}
     */
    et(this, "previous", /* @__PURE__ */ new Map());
    /**
     * When the batch is committed (and the DOM is updated), we need to remove old branches
     * and append new ones by calling the functions added inside (if/each/key/etc) blocks
     * @type {Set<() => void>}
     */
    X(this, Dt, /* @__PURE__ */ new Set());
    /**
     * If a fork is discarded, we need to destroy any effects that are no longer needed
     * @type {Set<(batch: Batch) => void>}
     */
    X(this, At, /* @__PURE__ */ new Set());
    /**
     * The number of async effects that are currently in flight
     */
    X(this, ft, 0);
    /**
     * The number of async effects that are currently in flight, _not_ inside a pending boundary
     */
    X(this, Lt, 0);
    /**
     * A deferred that resolves when the batch is committed, used with `settled()`
     * TODO replace with Promise.withResolvers once supported widely enough
     * @type {{ promise: Promise<void>, resolve: (value?: any) => void, reject: (reason: unknown) => void } | null}
     */
    X(this, jt, null);
    /**
     * Deferred effects (which run after async work has completed) that are DIRTY
     * @type {Set<Effect>}
     */
    X(this, Tt, /* @__PURE__ */ new Set());
    /**
     * Deferred effects that are MAYBE_DIRTY
     * @type {Set<Effect>}
     */
    X(this, St, /* @__PURE__ */ new Set());
    /**
     * A set of branches that still exist, but will be destroyed when this batch
     * is committed — we skip over these during `process`
     * @type {Set<Effect>}
     */
    et(this, "skipped_effects", /* @__PURE__ */ new Set());
    et(this, "is_fork", !1);
    X(this, Pt, !1);
  }
  is_deferred() {
    return this.is_fork || T(this, Lt) > 0;
  }
  /**
   *
   * @param {Effect[]} root_effects
   */
  process(t) {
    var s;
    z = [], this.apply();
    var n = [], i = [];
    for (const r of t)
      wt(this, J, _e).call(this, r, n, i);
    if (this.is_deferred())
      wt(this, J, Ce).call(this, i), wt(this, J, Ce).call(this, n);
    else {
      for (const r of T(this, Dt)) r();
      T(this, Dt).clear(), T(this, ft) === 0 && wt(this, J, pn).call(this), S = null, Fe(i), Fe(n), (s = T(this, jt)) == null || s.resolve();
    }
    R = null;
  }
  /**
   * Associate a change to a given source with the current
   * batch, noting its previous and current values
   * @param {Source} source
   * @param {any} value
   */
  capture(t, n) {
    n !== P && !this.previous.has(t) && this.previous.set(t, n), (t.f & st) === 0 && (this.current.set(t, t.v), R == null || R.set(t, t.v));
  }
  activate() {
    S = this, this.apply();
  }
  deactivate() {
    S === this && (S = null, R = null);
  }
  flush() {
    if (this.activate(), z.length > 0) {
      if (Ui(), S !== null && S !== this)
        return;
    } else T(this, ft) === 0 && this.process([]);
    this.deactivate();
  }
  discard() {
    for (const t of T(this, At)) t(this);
    T(this, At).clear();
  }
  /**
   *
   * @param {boolean} blocking
   */
  increment(t) {
    nt(this, ft, T(this, ft) + 1), t && nt(this, Lt, T(this, Lt) + 1);
  }
  /**
   *
   * @param {boolean} blocking
   */
  decrement(t) {
    nt(this, ft, T(this, ft) - 1), t && nt(this, Lt, T(this, Lt) - 1), !T(this, Pt) && (nt(this, Pt, !0), Bt(() => {
      nt(this, Pt, !1), this.is_deferred() ? z.length > 0 && this.flush() : this.revive();
    }));
  }
  revive() {
    for (const t of T(this, Tt))
      T(this, St).delete(t), x(t, j), Ot(t);
    for (const t of T(this, St))
      x(t, at), Ot(t);
    this.flush();
  }
  /** @param {() => void} fn */
  oncommit(t) {
    T(this, Dt).add(t);
  }
  /** @param {(batch: Batch) => void} fn */
  ondiscard(t) {
    T(this, At).add(t);
  }
  settled() {
    return (T(this, jt) ?? nt(this, jt, an())).promise;
  }
  static ensure() {
    if (S === null) {
      const t = S = new Re();
      $t.add(S), Bt(() => {
        S === t && t.flush();
      });
    }
    return S;
  }
  apply() {
  }
};
Dt = new WeakMap(), At = new WeakMap(), ft = new WeakMap(), Lt = new WeakMap(), jt = new WeakMap(), Tt = new WeakMap(), St = new WeakMap(), Pt = new WeakMap(), J = new WeakSet(), /**
 * Traverse the effect tree, executing effects or stashing
 * them for later execution as appropriate
 * @param {Effect} root
 * @param {Effect[]} effects
 * @param {Effect[]} render_effects
 */
_e = function(t, n, i) {
  t.f ^= U;
  for (var s = t.first, r = null; s !== null; ) {
    var o = s.f, a = (o & (mt | Rt)) !== 0, l = a && (o & U) !== 0, u = l || (o & Nt) !== 0 || this.skipped_effects.has(s);
    if (!u && s.fn !== null) {
      a ? s.f ^= U : r !== null && (o & (Qt | be | Ie)) !== 0 ? r.b.defer_effect(s) : (o & Qt) !== 0 ? n.push(s) : Gt(s) && ((o & gt) !== 0 && T(this, Tt).add(s), Wt(s));
      var c = s.first;
      if (c !== null) {
        s = c;
        continue;
      }
    }
    var d = s.parent;
    for (s = s.next; s === null && d !== null; )
      d === r && (r = null), s = d.next, d = d.parent;
  }
}, /**
 * @param {Effect[]} effects
 */
Ce = function(t) {
  for (var n = 0; n < t.length; n += 1)
    Mi(t[n], T(this, Tt), T(this, St));
}, pn = function() {
  var s;
  if ($t.size > 1) {
    this.previous.clear();
    var t = R, n = !0;
    for (const r of $t) {
      if (r === this) {
        n = !1;
        continue;
      }
      const o = [];
      for (const [l, u] of this.current) {
        if (r.current.has(l))
          if (n && u !== r.current.get(l))
            r.current.set(l, u);
          else
            continue;
        o.push(l);
      }
      if (o.length === 0)
        continue;
      const a = [...r.current.keys()].filter((l) => !this.current.has(l));
      if (a.length > 0) {
        var i = z;
        z = [];
        const l = /* @__PURE__ */ new Set(), u = /* @__PURE__ */ new Map();
        for (const c of o)
          _n(c, a, l, u);
        if (z.length > 0) {
          S = r, r.apply();
          for (const c of z)
            wt(s = r, J, _e).call(s, c, [], []);
          r.deactivate();
        }
        z = i;
      }
    }
    S = null, R = t;
  }
  this.committed = !0, $t.delete(this);
};
let te = Re;
function Ui() {
  pe = !0;
  var e = null;
  try {
    for (var t = 0; z.length > 0; ) {
      var n = te.ensure();
      if (t++ > 1e3) {
        var i, s;
        Fi();
      }
      n.process(z), rt.clear();
    }
  } finally {
    pe = !1, Ae = null;
  }
}
function Fi() {
  try {
    wi();
  } catch (e) {
    Jt(e, Ae);
  }
}
let K = null;
function Fe(e) {
  var t = e.length;
  if (t !== 0) {
    for (var n = 0; n < t; ) {
      var i = e[n++];
      if ((i.f & (tt | Nt)) === 0 && Gt(i) && (K = /* @__PURE__ */ new Set(), Wt(i), i.deps === null && i.first === null && i.nodes === null && (i.teardown === null && i.ac === null ? Tn(i) : i.fn = null), (K == null ? void 0 : K.size) > 0)) {
        rt.clear();
        for (const s of K) {
          if ((s.f & (tt | Nt)) !== 0) continue;
          const r = [s];
          let o = s.parent;
          for (; o !== null; )
            K.has(o) && (K.delete(o), r.push(o)), o = o.parent;
          for (let a = r.length - 1; a >= 0; a--) {
            const l = r[a];
            (l.f & (tt | Nt)) === 0 && Wt(l);
          }
        }
        K.clear();
      }
    }
    K = null;
  }
}
function _n(e, t, n, i) {
  if (!n.has(e) && (n.add(e), e.reactions !== null))
    for (const s of e.reactions) {
      const r = s.f;
      (r & k) !== 0 ? _n(
        /** @type {Derived} */
        s,
        t,
        n,
        i
      ) : (r & (Ne | gt)) !== 0 && (r & j) === 0 && Cn(s, t, i) && (x(s, j), Ot(
        /** @type {Effect} */
        s
      ));
    }
}
function Cn(e, t, n) {
  const i = n.get(e);
  if (i !== void 0) return i;
  if (e.deps !== null)
    for (const s of e.deps) {
      if (t.includes(s))
        return !0;
      if ((s.f & k) !== 0 && Cn(
        /** @type {Derived} */
        s,
        t,
        n
      ))
        return n.set(
          /** @type {Derived} */
          s,
          !0
        ), !0;
    }
  return n.set(e, !1), !1;
}
function Ot(e) {
  for (var t = Ae = e; t.parent !== null; ) {
    t = t.parent;
    var n = t.f;
    if (pe && t === D && (n & gt) !== 0 && (n & cn) === 0)
      return;
    if ((n & (Rt | mt)) !== 0) {
      if ((n & U) === 0) return;
      t.f ^= U;
    }
  }
  z.push(t);
}
function xi(e, t, n, i) {
  const s = Le;
  var r = e.filter((f) => !f.settled);
  if (n.length === 0 && r.length === 0) {
    i(t.map(s));
    return;
  }
  var o = S, a = (
    /** @type {Effect} */
    D
  ), l = Hi(), u = r.length === 1 ? r[0].promise : r.length > 1 ? Promise.all(r.map((f) => f.promise)) : null;
  function c(f) {
    l();
    try {
      i(f);
    } catch (p) {
      (a.f & tt) === 0 && Jt(p, a);
    }
    o == null || o.deactivate(), ye();
  }
  if (n.length === 0) {
    u.then(() => c(t.map(s)));
    return;
  }
  function d() {
    l(), Promise.all(n.map((f) => /* @__PURE__ */ Bi(f))).then((f) => c([...t.map(s), ...f])).catch((f) => Jt(f, a));
  }
  u ? u.then(d) : d();
}
function Hi() {
  var e = D, t = w, n = Q, i = S;
  return function(r = !0) {
    lt(e), Z(t), Zt(n), r && (i == null || i.activate());
  };
}
function ye() {
  lt(null), Z(null), Zt(null);
}
// @__NO_SIDE_EFFECTS__
function Le(e) {
  var t = k | j, n = w !== null && (w.f & k) !== 0 ? (
    /** @type {Derived} */
    w
  ) : null;
  return D !== null && (D.f |= se), {
    ctx: Q,
    deps: null,
    effects: null,
    equals: dn,
    f: t,
    fn: e,
    reactions: null,
    rv: 0,
    v: (
      /** @type {V} */
      P
    ),
    wv: 0,
    parent: n ?? D,
    ac: null
  };
}
// @__NO_SIDE_EFFECTS__
function Bi(e, t, n) {
  let i = (
    /** @type {Effect | null} */
    D
  );
  i === null && yi();
  var s = (
    /** @type {Boundary} */
    i.b
  ), r = (
    /** @type {Promise<V>} */
    /** @type {unknown} */
    void 0
  ), o = vn(
    /** @type {V} */
    P
  ), a = !w, l = /* @__PURE__ */ new Map();
  return ts(() => {
    var p;
    var u = an();
    r = u.promise;
    try {
      Promise.resolve(e()).then(u.resolve, u.reject).then(() => {
        c === S && c.committed && c.deactivate(), ye();
      });
    } catch (_) {
      u.reject(_), ye();
    }
    var c = (
      /** @type {Batch} */
      S
    );
    if (a) {
      var d = s.is_rendered();
      s.update_pending_count(1), c.increment(d), (p = l.get(c)) == null || p.reject(bt), l.delete(c), l.set(c, u);
    }
    const f = (_, m = void 0) => {
      if (c.activate(), m)
        m !== bt && (o.f |= st, me(o, m));
      else {
        (o.f & st) !== 0 && (o.f ^= st), me(o, _);
        for (const [v, y] of l) {
          if (l.delete(v), v === c) break;
          y.reject(bt);
        }
      }
      a && (s.update_pending_count(-1), c.decrement(d));
    };
    u.promise.then(f, (_) => f(null, _ || "unknown"));
  }), In(() => {
    for (const u of l.values())
      u.reject(bt);
  }), new Promise((u) => {
    function c(d) {
      function f() {
        d === r ? u(o) : c(r);
      }
      d.then(f, f);
    }
    c(r);
  });
}
// @__NO_SIDE_EFFECTS__
function Wi(e) {
  const t = /* @__PURE__ */ Le(e);
  return t.equals = Li, t;
}
function yn(e) {
  var t = e.effects;
  if (t !== null) {
    e.effects = null;
    for (var n = 0; n < t.length; n += 1)
      yt(
        /** @type {Effect} */
        t[n]
      );
  }
}
function Vi(e) {
  for (var t = e.parent; t !== null; ) {
    if ((t.f & k) === 0)
      return (t.f & tt) === 0 ? (
        /** @type {Effect} */
        t
      ) : null;
    t = t.parent;
  }
  return null;
}
function Te(e) {
  var t, n = D;
  lt(Vi(e));
  try {
    e.f &= ~Ct, yn(e), t = Rn(e);
  } finally {
    lt(n);
  }
  return t;
}
function gn(e) {
  var t = Te(e);
  if (!e.equals(t) && (e.wv = Pn(), (!(S != null && S.is_fork) || e.deps === null) && (e.v = t, e.deps === null))) {
    x(e, U);
    return;
  }
  ot || (R !== null ? (bn() || S != null && S.is_fork) && R.set(e, t) : De(e));
}
let ge = /* @__PURE__ */ new Set();
const rt = /* @__PURE__ */ new Map();
let mn = !1;
function vn(e, t) {
  var n = {
    f: 0,
    // TODO ideally we could skip this altogether, but it causes type errors
    v: e,
    reactions: null,
    equals: dn,
    rv: 0,
    wv: 0
  };
  return n;
}
// @__NO_SIDE_EFFECTS__
function $(e, t) {
  const n = vn(e);
  return ss(n), n;
}
function V(e, t, n = !1) {
  w !== null && // since we are untracking the function inside `$inspect.with` we need to add this check
  // to ensure we error if state is set inside an inspect effect
  (!q || (w.f & Ue) !== 0) && fn() && (w.f & (k | gt | Ne | Ue)) !== 0 && !(F != null && F.includes(e)) && Ni();
  let i = n ? Et(t) : t;
  return me(e, i);
}
function me(e, t) {
  if (!e.equals(t)) {
    var n = e.v;
    ot ? rt.set(e, t) : rt.set(e, n), e.v = t;
    var i = te.ensure();
    if (i.capture(e, n), (e.f & k) !== 0) {
      const s = (
        /** @type {Derived} */
        e
      );
      (e.f & j) !== 0 && Te(s), De(s);
    }
    e.wv = Pn(), wn(e, j), D !== null && (D.f & U) !== 0 && (D.f & (mt | Rt)) === 0 && (G === null ? rs([e]) : G.push(e)), !i.is_fork && ge.size > 0 && !mn && ji();
  }
  return t;
}
function ji() {
  mn = !1;
  for (const e of ge)
    (e.f & U) !== 0 && x(e, at), Gt(e) && Wt(e);
  ge.clear();
}
function oe(e) {
  V(e, e.v + 1);
}
function wn(e, t) {
  var n = e.reactions;
  if (n !== null)
    for (var i = n.length, s = 0; s < i; s++) {
      var r = n[s], o = r.f, a = (o & j) === 0;
      if (a && x(r, t), (o & k) !== 0) {
        var l = (
          /** @type {Derived} */
          r
        );
        R == null || R.delete(l), (o & Ct) === 0 && (o & Y && (r.f |= Ct), wn(l, at));
      } else a && ((o & gt) !== 0 && K !== null && K.add(
        /** @type {Effect} */
        r
      ), Ot(
        /** @type {Effect} */
        r
      ));
    }
}
function Et(e) {
  if (typeof e != "object" || e === null || ht in e)
    return e;
  const t = ln(e);
  if (t !== di && t !== fi)
    return e;
  var n = /* @__PURE__ */ new Map(), i = on(e), s = /* @__PURE__ */ $(0), r = pt, o = (a) => {
    if (pt === r)
      return a();
    var l = w, u = pt;
    Z(null), Be(r);
    var c = a();
    return Z(l), Be(u), c;
  };
  return i && n.set("length", /* @__PURE__ */ $(
    /** @type {any[]} */
    e.length
  )), new Proxy(
    /** @type {any} */
    e,
    {
      defineProperty(a, l, u) {
        (!("value" in u) || u.configurable === !1 || u.enumerable === !1 || u.writable === !1) && Ii();
        var c = n.get(l);
        return c === void 0 ? c = o(() => {
          var d = /* @__PURE__ */ $(u.value);
          return n.set(l, d), d;
        }) : V(c, u.value, !0), !0;
      },
      deleteProperty(a, l) {
        var u = n.get(l);
        if (u === void 0) {
          if (l in a) {
            const c = o(() => /* @__PURE__ */ $(P));
            n.set(l, c), oe(s);
          }
        } else
          V(u, P), oe(s);
        return !0;
      },
      get(a, l, u) {
        var p;
        if (l === ht)
          return e;
        var c = n.get(l), d = l in a;
        if (c === void 0 && (!d || (p = qt(a, l)) != null && p.writable) && (c = o(() => {
          var _ = Et(d ? a[l] : P), m = /* @__PURE__ */ $(_);
          return m;
        }), n.set(l, c)), c !== void 0) {
          var f = E(c);
          return f === P ? void 0 : f;
        }
        return Reflect.get(a, l, u);
      },
      getOwnPropertyDescriptor(a, l) {
        var u = Reflect.getOwnPropertyDescriptor(a, l);
        if (u && "value" in u) {
          var c = n.get(l);
          c && (u.value = E(c));
        } else if (u === void 0) {
          var d = n.get(l), f = d == null ? void 0 : d.v;
          if (d !== void 0 && f !== P)
            return {
              enumerable: !0,
              configurable: !0,
              value: f,
              writable: !0
            };
        }
        return u;
      },
      has(a, l) {
        var f;
        if (l === ht)
          return !0;
        var u = n.get(l), c = u !== void 0 && u.v !== P || Reflect.has(a, l);
        if (u !== void 0 || D !== null && (!c || (f = qt(a, l)) != null && f.writable)) {
          u === void 0 && (u = o(() => {
            var p = c ? Et(a[l]) : P, _ = /* @__PURE__ */ $(p);
            return _;
          }), n.set(l, u));
          var d = E(u);
          if (d === P)
            return !1;
        }
        return c;
      },
      set(a, l, u, c) {
        var h;
        var d = n.get(l), f = l in a;
        if (i && l === "length")
          for (var p = u; p < /** @type {Source<number>} */
          d.v; p += 1) {
            var _ = n.get(p + "");
            _ !== void 0 ? V(_, P) : p in a && (_ = o(() => /* @__PURE__ */ $(P)), n.set(p + "", _));
          }
        if (d === void 0)
          (!f || (h = qt(a, l)) != null && h.writable) && (d = o(() => /* @__PURE__ */ $(void 0)), V(d, Et(u)), n.set(l, d));
        else {
          f = d.v !== P;
          var m = o(() => Et(u));
          V(d, m);
        }
        var v = Reflect.getOwnPropertyDescriptor(a, l);
        if (v != null && v.set && v.set.call(c, u), !f) {
          if (i && typeof l == "string") {
            var y = (
              /** @type {Source<number>} */
              n.get("length")
            ), b = Number(l);
            Number.isInteger(b) && b >= y.v && V(y, b + 1);
          }
          oe(s);
        }
        return !0;
      },
      ownKeys(a) {
        E(s);
        var l = Reflect.ownKeys(a).filter((d) => {
          var f = n.get(d);
          return f === void 0 || f.v !== P;
        });
        for (var [u, c] of n)
          c.v !== P && !(u in a) && l.push(u);
        return l;
      },
      setPrototypeOf() {
        Ei();
      }
    }
  );
}
function xe(e) {
  try {
    if (e !== null && typeof e == "object" && ht in e)
      return e[ht];
  } catch {
  }
  return e;
}
function Gi(e, t) {
  return Object.is(xe(e), xe(t));
}
var $i, Yi, Ki;
// @__NO_SIDE_EFFECTS__
function zi(e) {
  return (
    /** @type {TemplateNode | null} */
    Yi.call(e)
  );
}
// @__NO_SIDE_EFFECTS__
function qi(e) {
  return (
    /** @type {TemplateNode | null} */
    Ki.call(e)
  );
}
function Xi(e, t) {
  if (t) {
    const n = document.body;
    e.autofocus = !0, Bt(() => {
      document.activeElement === n && e.focus();
    });
  }
}
function Se(e) {
  var t = w, n = D;
  Z(null), lt(null);
  try {
    return e();
  } finally {
    Z(t), lt(n);
  }
}
function Qi(e) {
  D === null && (w === null && vi(), mi()), ot && gi();
}
function Zi(e, t) {
  var n = t.last;
  n === null ? t.last = t.first = e : (n.next = e, e.prev = n, t.last = e);
}
function vt(e, t, n) {
  var i = D;
  i !== null && (i.f & Nt) !== 0 && (e |= Nt);
  var s = {
    ctx: Q,
    deps: null,
    nodes: null,
    f: e | j | Y,
    first: null,
    fn: t,
    last: null,
    next: null,
    parent: i,
    b: i && i.b,
    prev: null,
    teardown: null,
    wv: 0,
    ac: null
  };
  if (n)
    try {
      Wt(s), s.f |= Ee;
    } catch (a) {
      throw yt(s), a;
    }
  else t !== null && Ot(s);
  var r = s;
  if (n && r.deps === null && r.teardown === null && r.nodes === null && r.first === r.last && // either `null`, or a singular child
  (r.f & se) === 0 && (r = r.first, (e & gt) !== 0 && (e & Me) !== 0 && r !== null && (r.f |= Me)), r !== null && (r.parent = i, i !== null && Zi(r, i), w !== null && (w.f & k) !== 0 && (e & Rt) === 0)) {
    var o = (
      /** @type {Derived} */
      w
    );
    (o.effects ?? (o.effects = [])).push(r);
  }
  return s;
}
function bn() {
  return w !== null && !q;
}
function In(e) {
  const t = vt(be, null, !1);
  return x(t, U), t.teardown = e, t;
}
function Ji(e) {
  Qi();
  var t = (
    /** @type {Effect} */
    D.f
  ), n = !w && (t & mt) !== 0 && (t & Ee) === 0;
  if (n) {
    var i = (
      /** @type {ComponentContext} */
      Q
    );
    (i.e ?? (i.e = [])).push(e);
  } else
    return En(e);
}
function En(e) {
  return vt(Qt | pi, e, !1);
}
function Pe(e) {
  return vt(Qt, e, !1);
}
function ts(e) {
  return vt(Ne | se, e, !0);
}
function es(e, t = 0) {
  return vt(be | t, e, !0);
}
function Nn(e, t = 0) {
  var n = vt(Ie | t, e, !0);
  return n;
}
function Dn(e) {
  return vt(mt | se, e, !0);
}
function An(e) {
  var t = e.teardown;
  if (t !== null) {
    const n = ot, i = w;
    He(!0), Z(null);
    try {
      t.call(null);
    } finally {
      He(n), Z(i);
    }
  }
}
function Ln(e, t = !1) {
  var n = e.first;
  for (e.first = e.last = null; n !== null; ) {
    const s = n.ac;
    s !== null && Se(() => {
      s.abort(bt);
    });
    var i = n.next;
    (n.f & Rt) !== 0 ? n.parent = null : yt(n, t), n = i;
  }
}
function ns(e) {
  for (var t = e.first; t !== null; ) {
    var n = t.next;
    (t.f & mt) === 0 && yt(t), t = n;
  }
}
function yt(e, t = !0) {
  var n = !1;
  (t || (e.f & cn) !== 0) && e.nodes !== null && e.nodes.end !== null && (is(
    e.nodes.start,
    /** @type {TemplateNode} */
    e.nodes.end
  ), n = !0), Ln(e, t && !n), ee(e, 0), x(e, tt);
  var i = e.nodes && e.nodes.t;
  if (i !== null)
    for (const r of i)
      r.stop();
  An(e);
  var s = e.parent;
  s !== null && s.first !== null && Tn(e), e.next = e.prev = e.teardown = e.ctx = e.deps = e.fn = e.nodes = e.ac = null;
}
function is(e, t) {
  for (; e !== null; ) {
    var n = e === t ? null : /* @__PURE__ */ qi(e);
    e.remove(), e = n;
  }
}
function Tn(e) {
  var t = e.parent, n = e.prev, i = e.next;
  n !== null && (n.next = i), i !== null && (i.prev = n), t !== null && (t.first === e && (t.first = i), t.last === e && (t.last = n));
}
let Xt = !1, ot = !1;
function He(e) {
  ot = e;
}
let w = null, q = !1;
function Z(e) {
  w = e;
}
let D = null;
function lt(e) {
  D = e;
}
let F = null;
function ss(e) {
  w !== null && (F === null ? F = [e] : F.push(e));
}
let H = null, B = 0, G = null;
function rs(e) {
  G = e;
}
let Sn = 1, dt = 0, pt = dt;
function Be(e) {
  pt = e;
}
function Pn() {
  return ++Sn;
}
function Gt(e) {
  var t = e.f;
  if ((t & j) !== 0)
    return !0;
  if (t & k && (e.f &= ~Ct), (t & at) !== 0) {
    for (var n = (
      /** @type {Value[]} */
      e.deps
    ), i = n.length, s = 0; s < i; s++) {
      var r = n[s];
      if (Gt(
        /** @type {Derived} */
        r
      ) && gn(
        /** @type {Derived} */
        r
      ), r.wv > e.wv)
        return !0;
    }
    (t & Y) !== 0 && // During time traveling we don't want to reset the status so that
    // traversal of the graph in the other batches still happens
    R === null && x(e, U);
  }
  return !1;
}
function On(e, t, n = !0) {
  var i = e.reactions;
  if (i !== null && !(F != null && F.includes(e)))
    for (var s = 0; s < i.length; s++) {
      var r = i[s];
      (r.f & k) !== 0 ? On(
        /** @type {Derived} */
        r,
        t,
        !1
      ) : t === r && (n ? x(r, j) : (r.f & U) !== 0 && x(r, at), Ot(
        /** @type {Effect} */
        r
      ));
    }
}
function Rn(e) {
  var _;
  var t = H, n = B, i = G, s = w, r = F, o = Q, a = q, l = pt, u = e.f;
  H = /** @type {null | Value[]} */
  null, B = 0, G = null, w = (u & (mt | Rt)) === 0 ? e : null, F = null, Zt(e.ctx), q = !1, pt = ++dt, e.ac !== null && (Se(() => {
    e.ac.abort(bt);
  }), e.ac = null);
  try {
    e.f |= he;
    var c = (
      /** @type {Function} */
      e.fn
    ), d = c(), f = e.deps;
    if (H !== null) {
      var p;
      if (ee(e, B), f !== null && B > 0)
        for (f.length = B + H.length, p = 0; p < H.length; p++)
          f[B + p] = H[p];
      else
        e.deps = f = H;
      if (bn() && (e.f & Y) !== 0)
        for (p = B; p < f.length; p++)
          ((_ = f[p]).reactions ?? (_.reactions = [])).push(e);
    } else f !== null && B < f.length && (ee(e, B), f.length = B);
    if (fn() && G !== null && !q && f !== null && (e.f & (k | at | j)) === 0)
      for (p = 0; p < /** @type {Source[]} */
      G.length; p++)
        On(
          G[p],
          /** @type {Effect} */
          e
        );
    if (s !== null && s !== e) {
      if (dt++, s.deps !== null)
        for (let m = 0; m < n; m += 1)
          s.deps[m].rv = dt;
      if (t !== null)
        for (const m of t)
          m.rv = dt;
      G !== null && (i === null ? i = G : i.push(.../** @type {Source[]} */
      G));
    }
    return (e.f & st) !== 0 && (e.f ^= st), d;
  } catch (m) {
    return Ri(m);
  } finally {
    e.f ^= he, H = t, B = n, G = i, w = s, F = r, Zt(o), q = a, pt = l;
  }
}
function os(e, t) {
  let n = t.reactions;
  if (n !== null) {
    var i = ai.call(n, e);
    if (i !== -1) {
      var s = n.length - 1;
      s === 0 ? n = t.reactions = null : (n[i] = n[s], n.pop());
    }
  }
  if (n === null && (t.f & k) !== 0 && // Destroying a child effect while updating a parent effect can cause a dependency to appear
  // to be unused, when in fact it is used by the currently-updating parent. Checking `new_deps`
  // allows us to skip the expensive work of disconnecting and immediately reconnecting it
  (H === null || !H.includes(t))) {
    var r = (
      /** @type {Derived} */
      t
    );
    (r.f & Y) !== 0 && (r.f ^= Y, r.f &= ~Ct), De(r), yn(r), ee(r, 0);
  }
}
function ee(e, t) {
  var n = e.deps;
  if (n !== null)
    for (var i = t; i < n.length; i++)
      os(e, n[i]);
}
function Wt(e) {
  var t = e.f;
  if ((t & tt) === 0) {
    x(e, U);
    var n = D, i = Xt;
    D = e, Xt = !0;
    try {
      (t & (gt | Ie)) !== 0 ? ns(e) : Ln(e), An(e);
      var s = Rn(e);
      e.teardown = typeof s == "function" ? s : null, e.wv = Sn;
      var r;
      fe && Ti && (e.f & j) !== 0 && e.deps;
    } finally {
      Xt = i, D = n;
    }
  }
}
function E(e) {
  var t = e.f, n = (t & k) !== 0;
  if (w !== null && !q) {
    var i = D !== null && (D.f & tt) !== 0;
    if (!i && !(F != null && F.includes(e))) {
      var s = w.deps;
      if ((w.f & he) !== 0)
        e.rv < dt && (e.rv = dt, H === null && s !== null && s[B] === e ? B++ : H === null ? H = [e] : H.push(e));
      else {
        (w.deps ?? (w.deps = [])).push(e);
        var r = e.reactions;
        r === null ? e.reactions = [w] : r.includes(w) || r.push(w);
      }
    }
  }
  if (ot && rt.has(e))
    return rt.get(e);
  if (n) {
    var o = (
      /** @type {Derived} */
      e
    );
    if (ot) {
      var a = o.v;
      return ((o.f & U) === 0 && o.reactions !== null || Mn(o)) && (a = Te(o)), rt.set(o, a), a;
    }
    var l = (o.f & Y) === 0 && !q && w !== null && (Xt || (w.f & Y) !== 0), u = o.deps === null;
    Gt(o) && (l && (o.f |= Y), gn(o)), l && !u && kn(o);
  }
  if (R != null && R.has(e))
    return R.get(e);
  if ((e.f & st) !== 0)
    throw e.v;
  return e.v;
}
function kn(e) {
  if (e.deps !== null) {
    e.f |= Y;
    for (const t of e.deps)
      (t.reactions ?? (t.reactions = [])).push(e), (t.f & k) !== 0 && (t.f & Y) === 0 && kn(
        /** @type {Derived} */
        t
      );
  }
}
function Mn(e) {
  if (e.v === P) return !0;
  if (e.deps === null) return !1;
  for (const t of e.deps)
    if (rt.has(t) || (t.f & k) !== 0 && Mn(
      /** @type {Derived} */
      t
    ))
      return !0;
  return !1;
}
function Un(e) {
  var t = q;
  try {
    return q = !0, e();
  } finally {
    q = t;
  }
}
const ls = /* @__PURE__ */ new Set(), as = /* @__PURE__ */ new Set();
function us(e, t, n, i = {}) {
  function s(r) {
    if (i.capture || ds.call(t, r), !r.cancelBubble)
      return Se(() => n == null ? void 0 : n.call(this, r));
  }
  return e.startsWith("pointer") || e.startsWith("touch") || e === "wheel" ? Bt(() => {
    t.addEventListener(e, s, i);
  }) : t.addEventListener(e, s, i), s;
}
function cs(e) {
  for (var t = 0; t < e.length; t++)
    ls.add(e[t]);
  for (var n of as)
    n(e);
}
let We = null;
function ds(e) {
  var v;
  var t = this, n = (
    /** @type {Node} */
    t.ownerDocument
  ), i = e.type, s = ((v = e.composedPath) == null ? void 0 : v.call(e)) || [], r = (
    /** @type {null | Element} */
    s[0] || e.target
  );
  We = e;
  var o = 0, a = We === e && e.__root;
  if (a) {
    var l = s.indexOf(a);
    if (l !== -1 && (t === document || t === /** @type {any} */
    window)) {
      e.__root = t;
      return;
    }
    var u = s.indexOf(t);
    if (u === -1)
      return;
    l <= u && (o = l);
  }
  if (r = /** @type {Element} */
  s[o] || e.target, r !== t) {
    ui(e, "currentTarget", {
      configurable: !0,
      get() {
        return r || n;
      }
    });
    var c = w, d = D;
    Z(null), lt(null);
    try {
      for (var f, p = []; r !== null; ) {
        var _ = r.assignedSlot || r.parentNode || /** @type {any} */
        r.host || null;
        try {
          var m = r["__" + i];
          m != null && (!/** @type {any} */
          r.disabled || // DOM could've been updated already by the time this is reached, so we check this as well
          // -> the target could not have been disabled because it emits the event in the first place
          e.target === r) && m.call(r, e);
        } catch (y) {
          f ? p.push(y) : f = y;
        }
        if (e.cancelBubble || _ === t || _ === null)
          break;
        r = _;
      }
      if (f) {
        for (let y of p)
          queueMicrotask(() => {
            throw y;
          });
        throw f;
      }
    } finally {
      e.__root = t, delete e.currentTarget, Z(c), lt(d);
    }
  }
}
function fs(e) {
  var t = document.createElement("template");
  return t.innerHTML = e.replaceAll("<!>", "<!---->"), t.content;
}
function hs(e, t) {
  var n = (
    /** @type {Effect} */
    D
  );
  n.nodes === null && (n.nodes = { start: e, end: t, a: null, t: null });
}
// @__NO_SIDE_EFFECTS__
function ps(e, t) {
  var n = (t & ri) !== 0, i, s = !e.startsWith("<!>");
  return () => {
    i === void 0 && (i = fs(s ? e : "<!>" + e), i = /** @type {TemplateNode} */
    /* @__PURE__ */ zi(i));
    var r = (
      /** @type {TemplateNode} */
      n || $i ? document.importNode(i, !0) : i.cloneNode(!0)
    );
    return hs(r, r), r;
  };
}
function _s(e, t) {
  e !== null && e.before(
    /** @type {Node} */
    t
  );
}
function Cs(e) {
  return e.endsWith("capture") && e !== "gotpointercapture" && e !== "lostpointercapture";
}
const ys = [
  "beforeinput",
  "click",
  "change",
  "dblclick",
  "contextmenu",
  "focusin",
  "focusout",
  "input",
  "keydown",
  "keyup",
  "mousedown",
  "mousemove",
  "mouseout",
  "mouseover",
  "mouseup",
  "pointerdown",
  "pointermove",
  "pointerout",
  "pointerover",
  "pointerup",
  "touchend",
  "touchmove",
  "touchstart"
];
function gs(e) {
  return ys.includes(e);
}
const ms = {
  // no `class: 'className'` because we handle that separately
  formnovalidate: "formNoValidate",
  ismap: "isMap",
  nomodule: "noModule",
  playsinline: "playsInline",
  readonly: "readOnly",
  defaultvalue: "defaultValue",
  defaultchecked: "defaultChecked",
  srcobject: "srcObject",
  novalidate: "noValidate",
  allowfullscreen: "allowFullscreen",
  disablepictureinpicture: "disablePictureInPicture",
  disableremoteplayback: "disableRemotePlayback"
};
function vs(e) {
  return e = e.toLowerCase(), ms[e] ?? e;
}
function ws(e, t) {
  var n = void 0, i;
  Nn(() => {
    n !== (n = t()) && (i && (yt(i), i = null), n && (i = Dn(() => {
      Pe(() => (
        /** @type {(node: Element) => void} */
        n(e)
      ));
    })));
  });
}
function Fn(e) {
  var t, n, i = "";
  if (typeof e == "string" || typeof e == "number") i += e;
  else if (typeof e == "object") if (Array.isArray(e)) {
    var s = e.length;
    for (t = 0; t < s; t++) e[t] && (n = Fn(e[t])) && (i && (i += " "), i += n);
  } else for (n in e) e[n] && (i && (i += " "), i += n);
  return i;
}
function bs() {
  for (var e, t, n = 0, i = "", s = arguments.length; n < s; n++) (e = arguments[n]) && (t = Fn(e)) && (i && (i += " "), i += t);
  return i;
}
function Is(e) {
  return typeof e == "object" ? bs(e) : e ?? "";
}
const Ve = [...` 	
\r\f \v\uFEFF`];
function Es(e, t, n) {
  var i = e == null ? "" : "" + e;
  if (n) {
    for (var s in n)
      if (n[s])
        i = i ? i + " " + s : s;
      else if (i.length)
        for (var r = s.length, o = 0; (o = i.indexOf(s, o)) >= 0; ) {
          var a = o + r;
          (o === 0 || Ve.includes(i[o - 1])) && (a === i.length || Ve.includes(i[a])) ? i = (o === 0 ? "" : i.substring(0, o)) + i.substring(a + 1) : o = a;
        }
  }
  return i === "" ? null : i;
}
function je(e, t = !1) {
  var n = t ? " !important;" : ";", i = "";
  for (var s in e) {
    var r = e[s];
    r != null && r !== "" && (i += " " + s + ": " + r + n);
  }
  return i;
}
function le(e) {
  return e[0] !== "-" || e[1] !== "-" ? e.toLowerCase() : e;
}
function Ns(e, t) {
  if (t) {
    var n = "", i, s;
    if (Array.isArray(t) ? (i = t[0], s = t[1]) : i = t, e) {
      e = String(e).replaceAll(/\s*\/\*.*?\*\/\s*/g, "").trim();
      var r = !1, o = 0, a = !1, l = [];
      i && l.push(...Object.keys(i).map(le)), s && l.push(...Object.keys(s).map(le));
      var u = 0, c = -1;
      const m = e.length;
      for (var d = 0; d < m; d++) {
        var f = e[d];
        if (a ? f === "/" && e[d - 1] === "*" && (a = !1) : r ? r === f && (r = !1) : f === "/" && e[d + 1] === "*" ? a = !0 : f === '"' || f === "'" ? r = f : f === "(" ? o++ : f === ")" && o--, !a && r === !1 && o === 0) {
          if (f === ":" && c === -1)
            c = d;
          else if (f === ";" || d === m - 1) {
            if (c !== -1) {
              var p = le(e.substring(u, c).trim());
              if (!l.includes(p)) {
                f !== ";" && d++;
                var _ = e.substring(u, d).trim();
                n += " " + _ + ";";
              }
            }
            u = d + 1, c = -1;
          }
        }
      }
    }
    return i && (n += je(i)), s && (n += je(s, !0)), n = n.trim(), n === "" ? null : n;
  }
  return e == null ? null : String(e);
}
function Ds(e, t, n, i, s, r) {
  var o = e.__className;
  if (o !== n || o === void 0) {
    var a = Es(n, i, r);
    a == null ? e.removeAttribute("class") : t ? e.className = a : e.setAttribute("class", a), e.__className = n;
  } else if (r && s !== r)
    for (var l in r) {
      var u = !!r[l];
      (s == null || u !== !!s[l]) && e.classList.toggle(l, u);
    }
  return r;
}
function ae(e, t = {}, n, i) {
  for (var s in n) {
    var r = n[s];
    t[s] !== r && (n[s] == null ? e.style.removeProperty(s) : e.style.setProperty(s, r, i));
  }
}
function As(e, t, n, i) {
  var s = e.__style;
  if (s !== t) {
    var r = Ns(t, i);
    r == null ? e.removeAttribute("style") : e.style.cssText = r, e.__style = t;
  } else i && (Array.isArray(i) ? (ae(e, n == null ? void 0 : n[0], i[0]), ae(e, n == null ? void 0 : n[1], i[1], "important")) : ae(e, n, i));
  return i;
}
function ve(e, t, n = !1) {
  if (e.multiple) {
    if (t == null)
      return;
    if (!on(t))
      return Di();
    for (var i of e.options)
      i.selected = t.includes(Ge(i));
    return;
  }
  for (i of e.options) {
    var s = Ge(i);
    if (Gi(s, t)) {
      i.selected = !0;
      return;
    }
  }
  (!n || t !== void 0) && (e.selectedIndex = -1);
}
function Ls(e) {
  var t = new MutationObserver(() => {
    ve(e, e.__value);
  });
  t.observe(e, {
    // Listen to option element changes
    childList: !0,
    subtree: !0,
    // because of <optgroup>
    // Listen to option element value attribute changes
    // (doesn't get notified of select value changes,
    // because that property is not reflected as an attribute)
    attributes: !0,
    attributeFilter: ["value"]
  }), In(() => {
    t.disconnect();
  });
}
function Ge(e) {
  return "__value" in e ? e.__value : e.value;
}
const kt = Symbol("class"), Mt = Symbol("style"), xn = Symbol("is custom element"), Hn = Symbol("is html");
function Ts(e, t) {
  t ? e.hasAttribute("selected") || e.setAttribute("selected", "") : e.removeAttribute("selected");
}
function $e(e, t, n, i) {
  var s = Bn(e);
  s[t] !== (s[t] = n) && (t === "loading" && (e[Ci] = n), n == null ? e.removeAttribute(t) : typeof n != "string" && Wn(e).includes(t) ? e[t] = n : e.setAttribute(t, n));
}
function Ss(e, t, n, i, s = !1, r = !1) {
  var o = Bn(e), a = o[xn], l = !o[Hn], u = t || {}, c = e.tagName === "OPTION";
  for (var d in t)
    d in n || (n[d] = null);
  n.class ? n.class = Is(n.class) : n[kt] && (n.class = null), n[Mt] && (n.style ?? (n.style = null));
  var f = Wn(e);
  for (const h in n) {
    let g = n[h];
    if (c && h === "value" && g == null) {
      e.value = e.__value = "", u[h] = g;
      continue;
    }
    if (h === "class") {
      var p = e.namespaceURI === "http://www.w3.org/1999/xhtml";
      Ds(e, p, g, i, t == null ? void 0 : t[kt], n[kt]), u[h] = g, u[kt] = n[kt];
      continue;
    }
    if (h === "style") {
      As(e, g, t == null ? void 0 : t[Mt], n[Mt]), u[h] = g, u[Mt] = n[Mt];
      continue;
    }
    var _ = u[h];
    if (!(g === _ && !(g === void 0 && e.hasAttribute(h)))) {
      u[h] = g;
      var m = h[0] + h[1];
      if (m !== "$$")
        if (m === "on") {
          const L = {}, ut = "$$" + h;
          let O = h.slice(2);
          var v = gs(O);
          if (Cs(O) && (O = O.slice(0, -7), L.capture = !0), !v && _) {
            if (g != null) continue;
            e.removeEventListener(O, u[ut], L), u[ut] = null;
          }
          if (g != null)
            if (v)
              e[`__${O}`] = g, cs([O]);
            else {
              let zn = function(qn) {
                u[h].call(this, qn);
              };
              u[ut] = us(O, e, zn, L);
            }
          else v && (e[`__${O}`] = void 0);
        } else if (h === "style")
          $e(e, h, g);
        else if (h === "autofocus")
          Xi(
            /** @type {HTMLElement} */
            e,
            !!g
          );
        else if (!a && (h === "__value" || h === "value" && g != null))
          e.value = e.__value = g;
        else if (h === "selected" && c)
          Ts(
            /** @type {HTMLOptionElement} */
            e,
            g
          );
        else {
          var y = h;
          l || (y = vs(y));
          var b = y === "defaultValue" || y === "defaultChecked";
          if (g == null && !a && !b)
            if (o[h] = null, y === "value" || y === "checked") {
              let L = (
                /** @type {HTMLInputElement} */
                e
              );
              const ut = t === void 0;
              if (y === "value") {
                let O = L.defaultValue;
                L.removeAttribute(y), L.defaultValue = O, L.value = L.__value = ut ? O : null;
              } else {
                let O = L.defaultChecked;
                L.removeAttribute(y), L.defaultChecked = O, L.checked = ut ? O : !1;
              }
            } else
              e.removeAttribute(h);
          else b || f.includes(y) && (a || typeof g != "string") ? (e[y] = g, y in o && (o[y] = P)) : typeof g != "function" && $e(e, y, g);
        }
    }
  }
  return u;
}
function Ps(e, t, n = [], i = [], s = [], r, o = !1, a = !1) {
  xi(s, n, i, (l) => {
    var u = void 0, c = {}, d = e.nodeName === "SELECT", f = !1;
    if (Nn(() => {
      var _ = t(...l.map(E)), m = Ss(
        e,
        u,
        _,
        r,
        o,
        a
      );
      f && d && "value" in _ && ve(
        /** @type {HTMLSelectElement} */
        e,
        _.value
      );
      for (let y of Object.getOwnPropertySymbols(c))
        _[y] || yt(c[y]);
      for (let y of Object.getOwnPropertySymbols(_)) {
        var v = _[y];
        y.description === li && (!u || v !== u[y]) && (c[y] && yt(c[y]), c[y] = Dn(() => ws(e, () => v))), m[y] = v;
      }
      u = m;
    }), d) {
      var p = (
        /** @type {HTMLSelectElement} */
        e
      );
      Pe(() => {
        ve(
          p,
          /** @type {Record<string | symbol, any>} */
          u.value,
          !0
        ), Ls(p);
      });
    }
    f = !0;
  });
}
function Bn(e) {
  return (
    /** @type {Record<string | symbol, unknown>} **/
    // @ts-expect-error
    e.__attributes ?? (e.__attributes = {
      [xn]: e.nodeName.includes("-"),
      [Hn]: e.namespaceURI === oi
    })
  );
}
var Ye = /* @__PURE__ */ new Map();
function Wn(e) {
  var t = e.getAttribute("is") || e.nodeName, n = Ye.get(t);
  if (n) return n;
  Ye.set(t, n = []);
  for (var i, s = e, r = Element.prototype; r !== s; ) {
    i = ci(s);
    for (var o in i)
      i[o].set && n.push(o);
    s = ln(s);
  }
  return n;
}
function Ke(e, t) {
  return e === t || (e == null ? void 0 : e[ht]) === t;
}
function Os(e = {}, t, n, i) {
  return Pe(() => {
    var s, r;
    return es(() => {
      s = r, r = [], Un(() => {
        e !== n(...r) && (t(e, ...r), s && Ke(n(...s), e) && t(null, ...s));
      });
    }), () => {
      Bt(() => {
        r && Ke(n(...r), e) && t(null, ...r);
      });
    };
  }), e;
}
let Yt = !1;
function Rs(e) {
  var t = Yt;
  try {
    return Yt = !1, [e(), Yt];
  } finally {
    Yt = t;
  }
}
function Ut(e, t, n, i) {
  var b;
  var s = (n & ii) !== 0, r = (n & si) !== 0, o = (
    /** @type {V} */
    i
  ), a = !0, l = () => (a && (a = !1, o = r ? Un(
    /** @type {() => V} */
    i
  ) : (
    /** @type {V} */
    i
  )), o), u;
  if (s) {
    var c = ht in e || _i in e;
    u = ((b = qt(e, t)) == null ? void 0 : b.set) ?? (c && t in e ? (h) => e[t] = h : void 0);
  }
  var d, f = !1;
  s ? [d, f] = Rs(() => (
    /** @type {V} */
    e[t]
  )) : d = /** @type {V} */
  e[t], d === void 0 && i !== void 0 && (d = l(), u && (bi(), u(d)));
  var p;
  if (p = () => {
    var h = (
      /** @type {V} */
      e[t]
    );
    return h === void 0 ? l() : (a = !0, h);
  }, (n & ni) === 0)
    return p;
  if (u) {
    var _ = e.$$legacy;
    return (
      /** @type {() => V} */
      (function(h, g) {
        return arguments.length > 0 ? ((!g || _ || f) && u(g ? p() : h), h) : p();
      })
    );
  }
  var m = !1, v = ((n & ei) !== 0 ? Le : Wi)(() => (m = !1, p()));
  s && E(v);
  var y = (
    /** @type {Effect} */
    D
  );
  return (
    /** @type {() => V} */
    (function(h, g) {
      if (arguments.length > 0) {
        const L = g ? E(v) : s ? Et(h) : h;
        return V(v, L), m = !0, o !== void 0 && (o = L), h;
      }
      return ot && m || (y.f & tt) !== 0 ? v.v : E(v);
    })
  );
}
const ks = [
  [
    "af",
    // Afghanistan
    "93",
    0,
    null,
    "0"
  ],
  [
    "ax",
    // Åland Islands
    "358",
    1,
    ["18", "4"],
    // (4 is a mobile range shared with FI)
    "0"
  ],
  [
    "al",
    // Albania
    "355",
    0,
    null,
    "0"
  ],
  [
    "dz",
    // Algeria
    "213",
    0,
    null,
    "0"
  ],
  [
    "as",
    // American Samoa
    "1",
    5,
    ["684"],
    "1"
  ],
  [
    "ad",
    // Andorra
    "376"
  ],
  [
    "ao",
    // Angola
    "244"
  ],
  [
    "ai",
    // Anguilla
    "1",
    6,
    ["264"],
    "1"
  ],
  [
    "ag",
    // Antigua and Barbuda
    "1",
    7,
    ["268"],
    "1"
  ],
  [
    "ar",
    // Argentina
    "54",
    0,
    null,
    "0"
  ],
  [
    "am",
    // Armenia
    "374",
    0,
    null,
    "0"
  ],
  [
    "aw",
    // Aruba
    "297"
  ],
  [
    "ac",
    // Ascension Island
    "247"
  ],
  [
    "au",
    // Australia
    "61",
    0,
    ["4"],
    // (mobile range shared with CX and CC)
    "0"
  ],
  [
    "at",
    // Austria
    "43",
    0,
    null,
    "0"
  ],
  [
    "az",
    // Azerbaijan
    "994",
    0,
    null,
    "0"
  ],
  [
    "bs",
    // Bahamas
    "1",
    8,
    ["242"],
    "1"
  ],
  [
    "bh",
    // Bahrain
    "973"
  ],
  [
    "bd",
    // Bangladesh
    "880",
    0,
    null,
    "0"
  ],
  [
    "bb",
    // Barbados
    "1",
    9,
    ["246"],
    "1"
  ],
  [
    "by",
    // Belarus
    "375",
    0,
    null,
    "8"
  ],
  [
    "be",
    // Belgium
    "32",
    0,
    null,
    "0"
  ],
  [
    "bz",
    // Belize
    "501"
  ],
  [
    "bj",
    // Benin
    "229"
  ],
  [
    "bm",
    // Bermuda
    "1",
    10,
    ["441"],
    "1"
  ],
  [
    "bt",
    // Bhutan
    "975"
  ],
  [
    "bo",
    // Bolivia
    "591",
    0,
    null,
    "0"
  ],
  [
    "ba",
    // Bosnia and Herzegovina
    "387",
    0,
    null,
    "0"
  ],
  [
    "bw",
    // Botswana
    "267"
  ],
  [
    "br",
    // Brazil
    "55",
    0,
    null,
    "0"
  ],
  [
    "io",
    // British Indian Ocean Territory
    "246"
  ],
  [
    "vg",
    // British Virgin Islands
    "1",
    11,
    ["284"],
    "1"
  ],
  [
    "bn",
    // Brunei
    "673"
  ],
  [
    "bg",
    // Bulgaria
    "359",
    0,
    null,
    "0"
  ],
  [
    "bf",
    // Burkina Faso
    "226"
  ],
  [
    "bi",
    // Burundi
    "257"
  ],
  [
    "kh",
    // Cambodia
    "855",
    0,
    null,
    "0"
  ],
  [
    "cm",
    // Cameroon
    "237"
  ],
  [
    "ca",
    // Canada
    "1",
    1,
    ["204", "226", "236", "249", "250", "257", "263", "289", "306", "343", "354", "365", "367", "368", "382", "403", "416", "418", "428", "431", "437", "438", "450", "468", "474", "506", "514", "519", "548", "579", "581", "584", "587", "604", "613", "639", "647", "672", "683", "705", "709", "742", "753", "778", "780", "782", "807", "819", "825", "867", "873", "879", "902", "905", "942"],
    "1"
  ],
  [
    "cv",
    // Cape Verde
    "238"
  ],
  [
    "bq",
    // Caribbean Netherlands
    "599",
    1,
    ["3", "4", "7"]
  ],
  [
    "ky",
    // Cayman Islands
    "1",
    12,
    ["345"],
    "1"
  ],
  [
    "cf",
    // Central African Republic
    "236"
  ],
  [
    "td",
    // Chad
    "235"
  ],
  [
    "cl",
    // Chile
    "56"
  ],
  [
    "cn",
    // China
    "86",
    0,
    null,
    "0"
  ],
  [
    "cx",
    // Christmas Island
    "61",
    2,
    ["4", "89164"],
    // (4 is a mobile range shared with AU and CC)
    "0"
  ],
  [
    "cc",
    // Cocos (Keeling) Islands
    "61",
    1,
    ["4", "89162"],
    // (4 is a mobile range shared with AU and CX)
    "0"
  ],
  [
    "co",
    // Colombia
    "57",
    0,
    null,
    "0"
  ],
  [
    "km",
    // Comoros
    "269"
  ],
  [
    "cg",
    // Congo (Brazzaville)
    "242"
  ],
  [
    "cd",
    // Congo (Kinshasa)
    "243",
    0,
    null,
    "0"
  ],
  [
    "ck",
    // Cook Islands
    "682"
  ],
  [
    "cr",
    // Costa Rica
    "506"
  ],
  [
    "ci",
    // Côte d'Ivoire
    "225"
  ],
  [
    "hr",
    // Croatia
    "385",
    0,
    null,
    "0"
  ],
  [
    "cu",
    // Cuba
    "53",
    0,
    null,
    "0"
  ],
  [
    "cw",
    // Curaçao
    "599",
    0
  ],
  [
    "cy",
    // Cyprus
    "357"
  ],
  [
    "cz",
    // Czech Republic
    "420"
  ],
  [
    "dk",
    // Denmark
    "45"
  ],
  [
    "dj",
    // Djibouti
    "253"
  ],
  [
    "dm",
    // Dominica
    "1",
    13,
    ["767"],
    "1"
  ],
  [
    "do",
    // Dominican Republic
    "1",
    2,
    ["809", "829", "849"],
    "1"
  ],
  [
    "ec",
    // Ecuador
    "593",
    0,
    null,
    "0"
  ],
  [
    "eg",
    // Egypt
    "20",
    0,
    null,
    "0"
  ],
  [
    "sv",
    // El Salvador
    "503"
  ],
  [
    "gq",
    // Equatorial Guinea
    "240"
  ],
  [
    "er",
    // Eritrea
    "291",
    0,
    null,
    "0"
  ],
  [
    "ee",
    // Estonia
    "372"
  ],
  [
    "sz",
    // Eswatini
    "268"
  ],
  [
    "et",
    // Ethiopia
    "251",
    0,
    null,
    "0"
  ],
  [
    "fk",
    // Falkland Islands (Malvinas)
    "500"
  ],
  [
    "fo",
    // Faroe Islands
    "298"
  ],
  [
    "fj",
    // Fiji
    "679"
  ],
  [
    "fi",
    // Finland
    "358",
    0,
    ["4"],
    // (mobile range shared with AX)
    "0"
  ],
  [
    "fr",
    // France
    "33",
    0,
    null,
    "0"
  ],
  [
    "gf",
    // French Guiana
    "594",
    0,
    null,
    "0"
  ],
  [
    "pf",
    // French Polynesia
    "689"
  ],
  [
    "ga",
    // Gabon
    "241"
  ],
  [
    "gm",
    // Gambia
    "220"
  ],
  [
    "ge",
    // Georgia
    "995",
    0,
    null,
    "0"
  ],
  [
    "de",
    // Germany
    "49",
    0,
    null,
    "0"
  ],
  [
    "gh",
    // Ghana
    "233",
    0,
    null,
    "0"
  ],
  [
    "gi",
    // Gibraltar
    "350"
  ],
  [
    "gr",
    // Greece
    "30"
  ],
  [
    "gl",
    // Greenland
    "299"
  ],
  [
    "gd",
    // Grenada
    "1",
    14,
    ["473"],
    "1"
  ],
  [
    "gp",
    // Guadeloupe
    "590",
    0,
    null,
    "0"
  ],
  [
    "gu",
    // Guam
    "1",
    15,
    ["671"],
    "1"
  ],
  [
    "gt",
    // Guatemala
    "502"
  ],
  [
    "gg",
    // Guernsey
    "44",
    1,
    ["1481", "7781", "7839", "7911"],
    "0"
  ],
  [
    "gn",
    // Guinea
    "224"
  ],
  [
    "gw",
    // Guinea-Bissau
    "245"
  ],
  [
    "gy",
    // Guyana
    "592"
  ],
  [
    "ht",
    // Haiti
    "509"
  ],
  [
    "hn",
    // Honduras
    "504"
  ],
  [
    "hk",
    // Hong Kong SAR China
    "852"
  ],
  [
    "hu",
    // Hungary
    "36",
    0,
    null,
    "06"
  ],
  [
    "is",
    // Iceland
    "354"
  ],
  [
    "in",
    // India
    "91",
    0,
    null,
    "0"
  ],
  [
    "id",
    // Indonesia
    "62",
    0,
    null,
    "0"
  ],
  [
    "ir",
    // Iran
    "98",
    0,
    null,
    "0"
  ],
  [
    "iq",
    // Iraq
    "964",
    0,
    null,
    "0"
  ],
  [
    "ie",
    // Ireland
    "353",
    0,
    null,
    "0"
  ],
  [
    "im",
    // Isle of Man
    "44",
    2,
    ["1624", "74576", "7524", "7624", "7924"],
    "0"
  ],
  [
    "il",
    // Israel
    "972",
    0,
    null,
    "0"
  ],
  [
    "it",
    // Italy
    "39",
    0,
    ["3"]
    // (mobile range shared with VA)
  ],
  [
    "jm",
    // Jamaica
    "1",
    4,
    ["658", "876"],
    "1"
  ],
  [
    "jp",
    // Japan
    "81",
    0,
    null,
    "0"
  ],
  [
    "je",
    // Jersey
    "44",
    3,
    ["1534", "7509", "7700", "7797", "7829", "7937"],
    "0"
  ],
  [
    "jo",
    // Jordan
    "962",
    0,
    null,
    "0"
  ],
  [
    "kz",
    // Kazakhstan
    "7",
    1,
    ["33", "7"],
    // (33 is shared with RU)
    "8"
  ],
  [
    "ke",
    // Kenya
    "254",
    0,
    null,
    "0"
  ],
  [
    "ki",
    // Kiribati
    "686",
    0,
    null,
    "0"
  ],
  [
    "xk",
    // Kosovo
    "383",
    0,
    null,
    "0"
  ],
  [
    "kw",
    // Kuwait
    "965"
  ],
  [
    "kg",
    // Kyrgyzstan
    "996",
    0,
    null,
    "0"
  ],
  [
    "la",
    // Laos
    "856",
    0,
    null,
    "0"
  ],
  [
    "lv",
    // Latvia
    "371"
  ],
  [
    "lb",
    // Lebanon
    "961",
    0,
    null,
    "0"
  ],
  [
    "ls",
    // Lesotho
    "266"
  ],
  [
    "lr",
    // Liberia
    "231",
    0,
    null,
    "0"
  ],
  [
    "ly",
    // Libya
    "218",
    0,
    null,
    "0"
  ],
  [
    "li",
    // Liechtenstein
    "423",
    0,
    null,
    "0"
  ],
  [
    "lt",
    // Lithuania
    "370",
    0,
    null,
    "0"
  ],
  [
    "lu",
    // Luxembourg
    "352"
  ],
  [
    "mo",
    // Macao SAR China
    "853"
  ],
  [
    "mg",
    // Madagascar
    "261",
    0,
    null,
    "0"
  ],
  [
    "mw",
    // Malawi
    "265",
    0,
    null,
    "0"
  ],
  [
    "my",
    // Malaysia
    "60",
    0,
    null,
    "0"
  ],
  [
    "mv",
    // Maldives
    "960"
  ],
  [
    "ml",
    // Mali
    "223"
  ],
  [
    "mt",
    // Malta
    "356"
  ],
  [
    "mh",
    // Marshall Islands
    "692",
    0,
    null,
    "1"
  ],
  [
    "mq",
    // Martinique
    "596",
    0,
    null,
    "0"
  ],
  [
    "mr",
    // Mauritania
    "222"
  ],
  [
    "mu",
    // Mauritius
    "230"
  ],
  [
    "yt",
    // Mayotte
    "262",
    1,
    ["269", "639"],
    "0"
  ],
  [
    "mx",
    // Mexico
    "52"
  ],
  [
    "fm",
    // Micronesia
    "691"
  ],
  [
    "md",
    // Moldova
    "373",
    0,
    null,
    "0"
  ],
  [
    "mc",
    // Monaco
    "377",
    0,
    null,
    "0"
  ],
  [
    "mn",
    // Mongolia
    "976",
    0,
    null,
    "0"
  ],
  [
    "me",
    // Montenegro
    "382",
    0,
    null,
    "0"
  ],
  [
    "ms",
    // Montserrat
    "1",
    16,
    ["664"],
    "1"
  ],
  [
    "ma",
    // Morocco
    "212",
    0,
    ["6", "7"],
    // (mobile ranges shared with EH)
    "0"
  ],
  [
    "mz",
    // Mozambique
    "258"
  ],
  [
    "mm",
    // Myanmar (Burma)
    "95",
    0,
    null,
    "0"
  ],
  [
    "na",
    // Namibia
    "264",
    0,
    null,
    "0"
  ],
  [
    "nr",
    // Nauru
    "674"
  ],
  [
    "np",
    // Nepal
    "977",
    0,
    null,
    "0"
  ],
  [
    "nl",
    // Netherlands
    "31",
    0,
    null,
    "0"
  ],
  [
    "nc",
    // New Caledonia
    "687"
  ],
  [
    "nz",
    // New Zealand
    "64",
    0,
    null,
    "0"
  ],
  [
    "ni",
    // Nicaragua
    "505"
  ],
  [
    "ne",
    // Niger
    "227"
  ],
  [
    "ng",
    // Nigeria
    "234",
    0,
    null,
    "0"
  ],
  [
    "nu",
    // Niue
    "683"
  ],
  [
    "nf",
    // Norfolk Island
    "672"
  ],
  [
    "kp",
    // North Korea
    "850",
    0,
    null,
    "0"
  ],
  [
    "mk",
    // North Macedonia
    "389",
    0,
    null,
    "0"
  ],
  [
    "mp",
    // Northern Mariana Islands
    "1",
    17,
    ["670"],
    "1"
  ],
  [
    "no",
    // Norway
    "47",
    0,
    ["4", "9"]
    // (mobile ranges shared with SJ)
  ],
  [
    "om",
    // Oman
    "968"
  ],
  [
    "pk",
    // Pakistan
    "92",
    0,
    null,
    "0"
  ],
  [
    "pw",
    // Palau
    "680"
  ],
  [
    "ps",
    // Palestinian Territories
    "970",
    0,
    null,
    "0"
  ],
  [
    "pa",
    // Panama
    "507"
  ],
  [
    "pg",
    // Papua New Guinea
    "675"
  ],
  [
    "py",
    // Paraguay
    "595",
    0,
    null,
    "0"
  ],
  [
    "pe",
    // Peru
    "51",
    0,
    null,
    "0"
  ],
  [
    "ph",
    // Philippines
    "63",
    0,
    null,
    "0"
  ],
  [
    "pl",
    // Poland
    "48"
  ],
  [
    "pt",
    // Portugal
    "351"
  ],
  [
    "pr",
    // Puerto Rico
    "1",
    3,
    ["787", "939"],
    "1"
  ],
  [
    "qa",
    // Qatar
    "974"
  ],
  [
    "re",
    // Réunion
    "262",
    0,
    null,
    "0"
  ],
  [
    "ro",
    // Romania
    "40",
    0,
    null,
    "0"
  ],
  [
    "ru",
    // Russia
    "7",
    0,
    ["33"],
    // (shared with KZ)
    "8"
  ],
  [
    "rw",
    // Rwanda
    "250",
    0,
    null,
    "0"
  ],
  [
    "ws",
    // Samoa
    "685"
  ],
  [
    "sm",
    // San Marino
    "378"
  ],
  [
    "st",
    // São Tomé & Príncipe
    "239"
  ],
  [
    "sa",
    // Saudi Arabia
    "966",
    0,
    null,
    "0"
  ],
  [
    "sn",
    // Senegal
    "221"
  ],
  [
    "rs",
    // Serbia
    "381",
    0,
    null,
    "0"
  ],
  [
    "sc",
    // Seychelles
    "248"
  ],
  [
    "sl",
    // Sierra Leone
    "232",
    0,
    null,
    "0"
  ],
  [
    "sg",
    // Singapore
    "65"
  ],
  [
    "sx",
    // Sint Maarten
    "1",
    21,
    ["721"],
    "1"
  ],
  [
    "sk",
    // Slovakia
    "421",
    0,
    null,
    "0"
  ],
  [
    "si",
    // Slovenia
    "386",
    0,
    null,
    "0"
  ],
  [
    "sb",
    // Solomon Islands
    "677"
  ],
  [
    "so",
    // Somalia
    "252",
    0,
    null,
    "0"
  ],
  [
    "za",
    // South Africa
    "27",
    0,
    null,
    "0"
  ],
  [
    "kr",
    // South Korea
    "82",
    0,
    null,
    "0"
  ],
  [
    "ss",
    // South Sudan
    "211",
    0,
    null,
    "0"
  ],
  [
    "es",
    // Spain
    "34"
  ],
  [
    "lk",
    // Sri Lanka
    "94",
    0,
    null,
    "0"
  ],
  [
    "bl",
    // St. Barthélemy
    "590",
    1,
    null,
    "0"
  ],
  [
    "sh",
    // St. Helena
    "290"
  ],
  [
    "kn",
    // St. Kitts & Nevis
    "1",
    18,
    ["869"],
    "1"
  ],
  [
    "lc",
    // St. Lucia
    "1",
    19,
    ["758"],
    "1"
  ],
  [
    "mf",
    // St. Martin
    "590",
    2,
    null,
    "0"
  ],
  [
    "pm",
    // St. Pierre & Miquelon
    "508",
    0,
    null,
    "0"
  ],
  [
    "vc",
    // St. Vincent & Grenadines
    "1",
    20,
    ["784"],
    "1"
  ],
  [
    "sd",
    // Sudan
    "249",
    0,
    null,
    "0"
  ],
  [
    "sr",
    // Suriname
    "597"
  ],
  [
    "sj",
    // Svalbard & Jan Mayen
    "47",
    1,
    ["4", "79", "9"]
    // (4 and 9 are mobile ranges shared with NO)
  ],
  [
    "se",
    // Sweden
    "46",
    0,
    null,
    "0"
  ],
  [
    "ch",
    // Switzerland
    "41",
    0,
    null,
    "0"
  ],
  [
    "sy",
    // Syria
    "963",
    0,
    null,
    "0"
  ],
  [
    "tw",
    // Taiwan
    "886",
    0,
    null,
    "0"
  ],
  [
    "tj",
    // Tajikistan
    "992"
  ],
  [
    "tz",
    // Tanzania
    "255",
    0,
    null,
    "0"
  ],
  [
    "th",
    // Thailand
    "66",
    0,
    null,
    "0"
  ],
  [
    "tl",
    // Timor-Leste
    "670"
  ],
  [
    "tg",
    // Togo
    "228"
  ],
  [
    "tk",
    // Tokelau
    "690"
  ],
  [
    "to",
    // Tonga
    "676"
  ],
  [
    "tt",
    // Trinidad & Tobago
    "1",
    22,
    ["868"],
    "1"
  ],
  [
    "tn",
    // Tunisia
    "216"
  ],
  [
    "tr",
    // Turkey
    "90",
    0,
    null,
    "0"
  ],
  [
    "tm",
    // Turkmenistan
    "993",
    0,
    null,
    "8"
  ],
  [
    "tc",
    // Turks & Caicos Islands
    "1",
    23,
    ["649"],
    "1"
  ],
  [
    "tv",
    // Tuvalu
    "688"
  ],
  [
    "vi",
    // U.S. Virgin Islands
    "1",
    24,
    ["340"],
    "1"
  ],
  [
    "ug",
    // Uganda
    "256",
    0,
    null,
    "0"
  ],
  [
    "ua",
    // Ukraine
    "380",
    0,
    null,
    "0"
  ],
  [
    "ae",
    // United Arab Emirates
    "971",
    0,
    null,
    "0"
  ],
  [
    "gb",
    // United Kingdom
    "44",
    0,
    null,
    "0"
  ],
  [
    "us",
    // United States
    "1",
    0,
    null,
    "1"
  ],
  [
    "uy",
    // Uruguay
    "598",
    0,
    null,
    "0"
  ],
  [
    "uz",
    // Uzbekistan
    "998"
  ],
  [
    "vu",
    // Vanuatu
    "678"
  ],
  [
    "va",
    // Vatican City
    "39",
    1,
    ["06698", "3"]
    // (3 is a mobile range shared with IT)
  ],
  [
    "ve",
    // Venezuela
    "58",
    0,
    null,
    "0"
  ],
  [
    "vn",
    // Vietnam
    "84",
    0,
    null,
    "0"
  ],
  [
    "wf",
    // Wallis & Futuna
    "681"
  ],
  [
    "eh",
    // Western Sahara
    "212",
    1,
    ["5288", "5289", "6", "7"],
    // (6 and 7 are mobile ranges shared with MA)
    "0"
  ],
  [
    "ye",
    // Yemen
    "967",
    0,
    null,
    "0"
  ],
  [
    "zm",
    // Zambia
    "260",
    0,
    null,
    "0"
  ],
  [
    "zw",
    // Zimbabwe
    "263",
    0,
    null,
    "0"
  ]
], _t = [];
for (const e of ks)
  _t.push({
    name: "",
    // populated in the plugin
    iso2: e[0],
    dialCode: e[1],
    priority: e[2] || 0,
    areaCodes: e[3] || null,
    nodeById: {},
    // populated by the plugin
    nationalPrefix: e[4] || null,
    normalisedName: "",
    // populated in the plugin
    initials: "",
    // populated in the plugin
    dialCodePlus: ""
    // populated in the plugin
  });
const Kt = {
  OPEN_COUNTRY_DROPDOWN: "open:countrydropdown",
  CLOSE_COUNTRY_DROPDOWN: "close:countrydropdown",
  COUNTRY_CHANGE: "countrychange",
  INPUT: "input"
  // used for synthetic input trigger
}, I = {
  HIDE: "iti__hide",
  V_HIDE: "iti__v-hide",
  ARROW_UP: "iti__arrow--up",
  GLOBE: "iti__globe",
  FLAG: "iti__flag",
  LOADING: "iti__loading",
  COUNTRY_ITEM: "iti__country",
  HIGHLIGHT: "iti__highlight"
}, M = {
  ARROW_UP: "ArrowUp",
  ARROW_DOWN: "ArrowDown",
  SPACE: " ",
  ENTER: "Enter",
  ESC: "Escape",
  TAB: "Tab"
}, ze = {
  PASTE: "insertFromPaste",
  DELETE_FWD: "deleteContentForward"
}, ct = {
  ALPHA_UNICODE: new RegExp("\\p{L}", "u"),
  // any kind of letter from any language
  NON_PLUS_NUMERIC: /[^+0-9]/,
  // chars that are NOT + or digit
  NON_PLUS_NUMERIC_GLOBAL: /[^+0-9]/g,
  // chars that are NOT + or digit (global)
  HIDDEN_SEARCH_CHAR: /^[a-zA-ZÀ-ÿа-яА-Я ]$/
  // single acceptable hidden-search char
}, Vn = {
  SEARCH_DEBOUNCE_MS: 100,
  HIDDEN_SEARCH_RESET_MS: 1e3
}, qe = {
  UNKNOWN_NUMBER_TYPE: -99,
  UNKNOWN_VALIDATION_ERROR: -99
}, it = {
  NARROW_VIEWPORT_WIDTH: 500,
  // keep in sync with .iti__country-list CSS media query
  SANE_SELECTED_WITH_DIAL_WIDTH: 78,
  // px width fallback when separateDialCode enabled
  SANE_SELECTED_NO_DIAL_WIDTH: 42,
  // px width fallback when no separate dial code
  INPUT_PADDING_EXTRA_LEFT: 6,
  // px gap between selected country container and input text
  DROPDOWN_MARGIN: 3,
  // px margin between dropdown and tel input
  SANE_DROPDOWN_HEIGHT: 200
  // px height fallback for dropdown
}, jn = {
  NANP: "1"
  // North American Numbering Plan
}, ue = {
  DIAL_CODE: "44",
  // +44 United Kingdom
  MOBILE_PREFIX: "7",
  // UK mobile numbers start with 7 after national trunk (0) or core section
  MOBILE_CORE_LENGTH: 10
  // core number length (excluding dial code / national prefix) for mobiles
}, Ms = {
  ISO2: "us"
}, ne = {
  AGGRESSIVE: "aggressive",
  POLITE: "polite",
  OFF: "off"
}, Ft = {
  AUTO: "auto"
}, Us = [
  "FIXED_LINE",
  "MOBILE",
  "FIXED_LINE_OR_MOBILE",
  "TOLL_FREE",
  "PREMIUM_RATE",
  "SHARED_COST",
  "VOIP",
  "PERSONAL_NUMBER",
  "PAGER",
  "UAN",
  "VOICEMAIL",
  "UNKNOWN"
], zt = new Set(Us), Xe = {
  COUNTRY_CODE: "countryCode",
  DIAL_CODE: "dialCode"
}, N = {
  EXPANDED: "aria-expanded",
  LABEL: "aria-label",
  SELECTED: "aria-selected",
  ACTIVE_DESCENDANT: "aria-activedescendant",
  HASPOPUP: "aria-haspopup",
  CONTROLS: "aria-controls",
  HIDDEN: "aria-hidden",
  AUTOCOMPLETE: "aria-autocomplete",
  MODAL: "aria-modal"
}, Fs = {
  selectedCountryAriaLabel: "Change country, selected ${countryName} (${dialCode})",
  noCountrySelected: "Select country",
  countryListAriaLabel: "List of countries",
  searchPlaceholder: "Search",
  clearSearchAriaLabel: "Clear search",
  searchEmptyState: "No results found",
  searchSummaryAria(e) {
    return e === 0 ? "No results found" : e === 1 ? "1 result found" : `${e} results found`;
  }
}, we = (e) => typeof window < "u" && typeof window.matchMedia == "function" && window.matchMedia(e).matches, Gn = () => we(`(max-width: ${it.NARROW_VIEWPORT_WIDTH}px)`), xs = () => {
  if (typeof navigator < "u" && typeof window < "u") {
    const e = we("(max-height: 600px)"), t = we("(pointer: coarse)");
    return Gn() || t && e;
  }
  return !1;
}, Oe = {
  //* Whether or not to allow the dropdown.
  allowDropdown: !0,
  //* The number type to enforce during validation.
  allowedNumberTypes: ["MOBILE", "FIXED_LINE"],
  //* Whether or not to allow extensions after the main number.
  allowNumberExtensions: !1,
  // Allow alphanumeric "phonewords" (e.g. +1 800 FLOWERS) as valid numbers
  allowPhonewords: !1,
  //* Add a placeholder in the input with an example number for the selected country.
  autoPlaceholder: ne.POLITE,
  //* Add a custom class to the (injected) container element.
  containerClass: "",
  //* Locale for localising country names via Intl.DisplayNames.
  countryNameLocale: "en",
  //* The order of the countries in the dropdown. Defaults to alphabetical.
  countryOrder: null,
  //* Add a country search input at the top of the dropdown.
  countrySearch: !0,
  //* Modify the auto placeholder.
  customPlaceholder: null,
  //* Always show the dropdown
  dropdownAlwaysOpen: !1,
  //* Append menu to specified element.
  dropdownContainer: null,
  //* Don't display these countries.
  excludeCountries: [],
  //* Fix the dropdown width to the input width (rather than being as wide as the longest country name).
  fixDropdownWidth: !0,
  //* Format the number as the user types
  formatAsYouType: !0,
  //* Format the input value during initialisation and on setNumber.
  formatOnDisplay: !0,
  //* geoIp lookup function.
  geoIpLookup: null,
  //* Inject a hidden input with the name returned from this function, and on submit, populate it with the result of getNumber.
  hiddenInput: null,
  //* Internationalise the plugin text e.g. search input placeholder, country names.
  i18n: {},
  //* Initial country.
  initialCountry: "",
  //* A function to load the utils script.
  loadUtils: null,
  //* National vs international formatting for numbers e.g. placeholders and displaying existing numbers.
  nationalMode: !0,
  //* Display only these countries.
  onlyCountries: [],
  //* Number type to use for placeholders.
  placeholderNumberType: "MOBILE",
  //* Add custom classes to the search input element.
  searchInputClass: "",
  //* Display the international dial code next to the selected flag.
  separateDialCode: !1,
  //* Show flags - for both the selected country, and in the country dropdown
  showFlags: !0,
  //* Only allow certain chars e.g. a plus followed by numeric digits, and cap at max valid length.
  strictMode: !1,
  //* Use full screen popup instead of dropdown for country list.
  useFullscreenPopup: xs()
}, $n = (e) => JSON.stringify(e), Qe = (e) => !!e && typeof e == "object" && !Array.isArray(e), Hs = (e) => typeof e == "function", Bs = (e) => {
  if (!e || typeof e != "object") return !1;
  const t = e;
  return t.nodeType === 1 && typeof t.tagName == "string" && typeof t.appendChild == "function";
}, Ws = new Set(_t.map((e) => e.iso2)), Yn = (e) => Ws.has(e), Ze = new Set(Object.values(ne)), ie = (e) => {
  console.warn(`[intl-tel-input] ${e}`);
}, W = (e, t, n) => {
  ie(`Option '${e}' must be ${t}; got ${$n(n)}. Ignoring.`);
}, Vs = (e, t) => Object.prototype.hasOwnProperty.call(e, t), Je = (e, t) => {
  const n = "an array of ISO2 country code strings";
  if (!Array.isArray(t))
    return W(e, n, t), !1;
  for (const i of t) {
    if (typeof i != "string")
      return W(e, n, t), !1;
    const s = i.toLowerCase();
    if (!Yn(s))
      return ie(`Invalid country code in '${e}': '${i}'. Ignoring.`), !1;
  }
  return !0;
}, js = (e) => {
  if (e === void 0)
    return {};
  if (!Qe(e)) {
    const n = `The second argument must be an options object; got ${$n(e)}. Using defaults.`;
    return ie(n), {};
  }
  const t = {};
  for (const [n, i] of Object.entries(e)) {
    if (!Vs(Oe, n)) {
      ie(`Unknown option '${n}'. Ignoring.`);
      continue;
    }
    switch (n) {
      case "allowDropdown":
      case "allowNumberExtensions":
      case "allowPhonewords":
      case "countrySearch":
      case "dropdownAlwaysOpen":
      case "fixDropdownWidth":
      case "formatAsYouType":
      case "formatOnDisplay":
      case "nationalMode":
      case "showFlags":
      case "separateDialCode":
      case "strictMode":
      case "useFullscreenPopup":
        if (typeof i != "boolean") {
          W(n, "a boolean", i);
          break;
        }
        t[n] = i;
        break;
      case "autoPlaceholder":
        if (typeof i != "string" || !Ze.has(i)) {
          const s = Array.from(Ze).join(", ");
          W("autoPlaceholder", `one of ${s}`, i);
          break;
        }
        t[n] = i;
        break;
      case "containerClass":
      case "searchInputClass":
      case "countryNameLocale":
        if (typeof i != "string") {
          W(n, "a string", i);
          break;
        }
        t[n] = i;
        break;
      case "countryOrder":
        (i === null || Je(n, i)) && (t[n] = i);
        break;
      case "customPlaceholder":
      case "geoIpLookup":
      case "hiddenInput":
      case "loadUtils":
        if (i !== null && !Hs(i)) {
          W(n, "a function or null", i);
          break;
        }
        t[n] = i;
        break;
      case "dropdownContainer":
        if (i !== null && !Bs(i)) {
          W("dropdownContainer", "an HTMLElement or null", i);
          break;
        }
        t[n] = i;
        break;
      case "excludeCountries":
      case "onlyCountries":
        Je(n, i) && (t[n] = i);
        break;
      case "i18n":
        if (i && !Qe(i)) {
          W("i18n", "an object", i);
          break;
        }
        t[n] = i;
        break;
      case "initialCountry": {
        if (typeof i != "string") {
          W("initialCountry", "a string", i);
          break;
        }
        const s = i.toLowerCase();
        if (s && s !== Ft.AUTO && !Yn(s)) {
          W("initialCountry", "a valid ISO2 country code or 'auto'", i);
          break;
        }
        t[n] = i;
        break;
      }
      case "placeholderNumberType":
        if (typeof i != "string" || !zt.has(i)) {
          const s = Array.from(zt).join(", ");
          W("placeholderNumberType", `one of ${s}`, i);
          break;
        }
        t[n] = i;
        break;
      case "allowedNumberTypes":
        if (i !== null) {
          if (!Array.isArray(i)) {
            W("allowedNumberTypes", "an array of number types or null", i);
            break;
          }
          let s = !0;
          for (const r of i)
            if (typeof r != "string" || !zt.has(r)) {
              const o = Array.from(zt).join(", ");
              W("allowedNumberTypes", `an array of valid number types (${o})`, r), s = !1;
              break;
            }
          s && (t[n] = i);
        } else
          t[n] = null;
        break;
    }
  }
  return t;
}, Gs = (e) => {
  e.dropdownAlwaysOpen && (e.useFullscreenPopup = !1, e.allowDropdown = !0), e.useFullscreenPopup ? e.fixDropdownWidth = !1 : Gn() && (e.fixDropdownWidth = !0), e.onlyCountries.length === 1 && (e.initialCountry = e.onlyCountries[0]), e.separateDialCode && (e.nationalMode = !1), e.allowDropdown && !e.showFlags && !e.separateDialCode && (e.nationalMode = !1), e.useFullscreenPopup && !e.dropdownContainer && (e.dropdownContainer = document.body), e.i18n = { ...Fs, ...e.i18n };
}, xt = (e) => e.replace(/\D/g, ""), Kn = (e = "") => e.normalize("NFD").replace(/[\u0300-\u036f]/g, "").toLowerCase(), $s = () => typeof navigator < "u" ? /Android/i.test(navigator.userAgent) : !1, Ys = (e, t) => {
  const n = Kn(t), i = [], s = [], r = [], o = [], a = [], l = [];
  for (const c of e)
    c.iso2 === n ? i.push(c) : c.normalisedName.startsWith(n) ? s.push(c) : c.normalisedName.includes(n) ? r.push(c) : n === c.dialCode || n === c.dialCodePlus ? o.push(c) : c.dialCodePlus.includes(n) ? a.push(c) : c.initials.includes(n) && l.push(c);
  const u = (c, d) => c.priority - d.priority;
  return [
    ...i.sort(u),
    ...s.sort(u),
    ...r.sort(u),
    ...o.sort(u),
    ...a.sort(u),
    ...l.sort(u)
  ];
}, Ks = (e, t) => {
  const n = t.toLowerCase();
  for (const i of e)
    if (i.name.toLowerCase().startsWith(n))
      return i;
  return null;
}, ce = (e) => Object.keys(e).filter((t) => !!e[t]).join(" "), A = (e, t, n) => {
  const i = document.createElement(e);
  return t && Object.entries(t).forEach(
    ([s, r]) => i.setAttribute(s, r)
  ), n && n.appendChild(i), i;
}, zs = () => `
  <svg class="iti__search-icon-svg" width="14" height="14" viewBox="0 0 24 24" focusable="false" ${N.HIDDEN}="true">
    <circle cx="11" cy="11" r="7" />
    <line x1="21" y1="21" x2="16.65" y2="16.65" />
  </svg>`, qs = (e) => {
  const t = `iti-${e}-clear-mask`;
  return `
    <svg class="iti__search-clear-svg" width="12" height="12" viewBox="0 0 16 16" ${N.HIDDEN}="true" focusable="false">
      <mask id="${t}" maskUnits="userSpaceOnUse">
        <rect width="16" height="16" fill="white" />
        <path d="M5.2 5.2 L10.8 10.8 M10.8 5.2 L5.2 10.8" stroke="black" stroke-linecap="round" class="iti__search-clear-x" />
      </mask>
      <circle cx="8" cy="8" r="8" class="iti__search-clear-bg" mask="url(#${t})" />
    </svg>`;
}, Xs = () => `
  <svg class="iti__country-check-svg" width="14" height="14" viewBox="0 0 16 16" fill="currentColor" focusable="false" ${N.HIDDEN}="true">
    <path d="M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0m-3.97-3.03a.75.75 0 0 0-1.08.022L7.477 9.417 5.384 7.323a.75.75 0 0 0-1.06 1.06L6.97 11.03a.75.75 0 0 0 1.079-.02l3.992-4.99a.75.75 0 0 0-.01-1.05z"/>
  </svg>`, tn = () => `
  <svg width="256" height="256" viewBox="0 0 512 512" class="iti__globe-svg">
    <path d="M508 213a240 240 0 0 0-449-87l-2 5-2 5c-8 14-13 30-17 46a65 65 0 0 1 56 4c16-10 35-19 56-27l9-3c-6 23-10 48-10 74h-16l4 6c3 4 5 8 6 13h6c0 22 3 44 8 65l2 10-25-10-4 5 12 18 9 3 6 2 8 3 9 26 1 2 16-7h1l-5-13-1-2c24 6 49 9 75 10v26l11 10 7 7v-30l1-13c22 0 44-3 65-8l10-2-21 48-1 1a317 317 0 0 1-14 23l-21 5h-2c6 16 7 33 1 50a240 240 0 0 0 211-265m-401-56-11 6c19-44 54-79 98-98-11 20-21 44-29 69-21 6-40 15-58 23m154 182v4c-29-1-57-6-81-13-7-25-12-52-13-81h94zm0-109h-94c1-29 6-56 13-81 24-7 52-12 81-13zm0-112c-22 1-44 4-65 8l-10 2 12-30 9-17 1-2a332 332 0 0 1 13-23c13-4 26-6 40-7zm187 69 6 4c4 12 6 25 6 38v1h-68c-1-26-4-51-10-74l48 20 1 1 14 8zm-14-44 10 20c-20-11-43-21-68-29-8-25-18-49-29-69 37 16 67 44 87 78M279 49h1c13 1 27 3 39 7l14 23 1 2a343 343 0 0 1 12 26l2 5 6 16c-23-6-48-9-74-10h-1zm0 87h1c29 1 56 6 81 13 7 24 12 51 12 80v1h-94zm2 207h-2v-94h95c-1 29-6 56-13 81-24 7-51 12-80 13m86 60-20 10c11-20 21-43 29-68 25-8 48-18 68-29-16 37-43 67-77 87m87-115-7 5-16 9-2 1a337 337 0 0 1-47 21c6-24 9-49 10-75h68c0 13-2 27-6 39"/>
    <path d="m261 428-2-2-22-21a40 40 0 0 0-32-11h-1a37 37 0 0 0-18 8l-1 1-4 2-2 2-5 4c-9-3-36-31-47-44s-32-45-34-55l3-2a151 151 0 0 0 11-9v-1a39 39 0 0 0 5-48l-3-3-11-19-3-4-5-7h-1l-3-3-4-3-5-2a35 35 0 0 0-16-3h-5c-4 1-14 5-24 11l-4 2-4 3-4 2c-9 8-17 17-18 27a380 380 0 0 0 212 259h3c12 0 25-10 36-21l10-12 6-11a39 39 0 0 0-8-40"/>
  </svg>`;
class Vt {
  constructor(t, n, i) {
    this.searchKeyupTimer = null, this.inlineDropdownHeight = null, this.dropdownForContainer = null, this.highlightedItem = null, this.selectedItem = null, t.dataset.intlTelInputId = i.toString(), this.telInput = t, this.options = n, this.id = i, this.hadInitialPlaceholder = !!t.getAttribute("placeholder"), this.isRTL = !!this.telInput.closest("[dir=rtl]"), this.options.separateDialCode && (this.originalPaddingLeft = this.telInput.style.paddingLeft);
  }
  // Validate that the provided element is an HTMLInputElement.
  static validateInput(t) {
    const n = t == null ? void 0 : t.tagName;
    if (!(!!t && typeof t == "object" && n === "INPUT" && typeof t.setAttribute == "function")) {
      const s = Object.prototype.toString.call(t);
      throw new TypeError(
        `The first argument must be an HTMLInputElement, not ${s}`
      );
    }
  }
  //* Generate all of the markup for the plugin: the selected country overlay, and the dropdown.
  generateMarkup(t) {
    this.countries = t, this.telInput.classList.add("iti__tel-input"), this.telInput.hasAttribute("autocomplete") || this.telInput.setAttribute("autocomplete", "tel"), this.telInput.hasAttribute("inputmode") || this.telInput.setAttribute("inputmode", "tel");
    const n = this._createWrapperAndInsert();
    this._maybeBuildCountryContainer(n), n.appendChild(this.telInput), this._maybeUpdateInputPaddingAndReveal(), this._maybeBuildHiddenInputs(n);
  }
  _createWrapperAndInsert() {
    const {
      allowDropdown: t,
      showFlags: n,
      containerClass: i,
      useFullscreenPopup: s
    } = this.options, r = ce({
      iti: !0,
      "iti--allow-dropdown": t,
      "iti--show-flags": n,
      "iti--inline-dropdown": !s,
      [i]: !!i
    }), o = A("div", { class: r });
    return this.isRTL && o.setAttribute("dir", "ltr"), this.telInput.before(o), o;
  }
  _maybeBuildCountryContainer(t) {
    const { allowDropdown: n, separateDialCode: i, showFlags: s } = this.options;
    if (n || s || i) {
      this.countryContainer = A(
        "div",
        // visibly hidden until we measure it's width to set the input padding correctly
        { class: `iti__country-container ${I.V_HIDE}` },
        t
      ), n ? (this.selectedCountry = A(
        "button",
        {
          type: "button",
          class: "iti__selected-country",
          [N.EXPANDED]: "false",
          [N.LABEL]: this.options.i18n.noCountrySelected,
          [N.HASPOPUP]: "dialog",
          [N.CONTROLS]: `iti-${this.id}__dropdown-content`
        },
        this.countryContainer
      ), this.telInput.disabled && this.selectedCountry.setAttribute("disabled", "true")) : this.selectedCountry = A(
        "div",
        { class: "iti__selected-country" },
        this.countryContainer
      );
      const r = A(
        "div",
        { class: "iti__selected-country-primary" },
        this.selectedCountry
      );
      this.selectedCountryInner = A(
        "div",
        { class: I.FLAG },
        r
      ), n && (this.dropdownArrow = A(
        "div",
        { class: "iti__arrow", [N.HIDDEN]: "true" },
        r
      )), i && (this.selectedDialCode = A(
        "div",
        { class: "iti__selected-dial-code" },
        this.selectedCountry
      )), n && this._buildDropdownContent();
    }
  }
  _maybeEnsureDropdownWidthSet() {
    const { fixDropdownWidth: t } = this.options;
    if (t && !this.dropdownContent.style.width) {
      const n = this.telInput.offsetWidth;
      n > 0 && (this.dropdownContent.style.width = `${n}px`);
    }
  }
  _buildDropdownContent() {
    const {
      fixDropdownWidth: t,
      useFullscreenPopup: n,
      countrySearch: i,
      i18n: s,
      dropdownContainer: r,
      containerClass: o
    } = this.options, a = t ? "" : "iti--flexible-dropdown-width";
    if (this.dropdownContent = A("div", {
      id: `iti-${this.id}__dropdown-content`,
      class: `iti__dropdown-content ${I.HIDE} ${a}`,
      role: "dialog",
      [N.MODAL]: "true"
    }), this.isRTL && this.dropdownContent.setAttribute("dir", "rtl"), i && this._buildSearchUI(), this.countryList = A(
      "ul",
      {
        class: "iti__country-list",
        id: `iti-${this.id}__country-listbox`,
        role: "listbox",
        [N.LABEL]: s.countryListAriaLabel
      },
      this.dropdownContent
    ), this._appendListItems(), i && this.updateSearchResultsA11yText(), n || (this._maybeEnsureDropdownWidthSet(), this.inlineDropdownHeight = this.getHiddenInlineDropdownHeight(), i && (this.dropdownContent.style.height = `${this.inlineDropdownHeight}px`)), r) {
      const l = ce({
        iti: !0,
        "iti--container": !0,
        "iti--fullscreen-popup": n,
        "iti--inline-dropdown": !n,
        [o]: !!o
      });
      this.dropdownForContainer = A("div", { class: l }), this.dropdownForContainer.appendChild(this.dropdownContent);
    } else
      this.countryContainer.appendChild(this.dropdownContent);
  }
  _buildSearchUI() {
    const { i18n: t, searchInputClass: n } = this.options, i = A(
      "div",
      { class: "iti__search-input-wrapper" },
      this.dropdownContent
    );
    this.searchIcon = A(
      "span",
      {
        class: "iti__search-icon",
        [N.HIDDEN]: "true"
      },
      i
    ), this.searchIcon.innerHTML = zs(), this.searchInput = A(
      "input",
      {
        id: `iti-${this.id}__search-input`,
        // Chrome says inputs need either a name or an id
        type: "search",
        class: `iti__search-input ${n}`,
        placeholder: t.searchPlaceholder,
        // role=combobox + aria-autocomplete=list + aria-activedescendant allows maintaining focus on the search input while allowing users to navigate search results with up/down keyboard keys
        role: "combobox",
        [N.EXPANDED]: "true",
        [N.LABEL]: t.searchPlaceholder,
        [N.CONTROLS]: `iti-${this.id}__country-listbox`,
        [N.AUTOCOMPLETE]: "list",
        autocomplete: "off"
      },
      i
    ), this.searchClearButton = A(
      "button",
      {
        type: "button",
        class: `iti__search-clear ${I.HIDE}`,
        [N.LABEL]: t.clearSearchAriaLabel,
        tabindex: "-1"
      },
      i
    ), this.searchClearButton.innerHTML = qs(this.id), this.searchResultsA11yText = A(
      "span",
      { class: "iti__a11y-text" },
      this.dropdownContent
    ), this.searchNoResults = A(
      "div",
      {
        class: `iti__no-results ${I.HIDE}`,
        [N.HIDDEN]: "true"
        // all a11y messaging happens in this.searchResultsA11yText
      },
      this.dropdownContent
    ), this.searchNoResults.textContent = t.searchEmptyState;
  }
  _maybeUpdateInputPaddingAndReveal() {
    this.countryContainer && (this.updateInputPadding(), this.countryContainer.classList.remove(I.V_HIDE));
  }
  _maybeBuildHiddenInputs(t) {
    var i, s;
    const { hiddenInput: n } = this.options;
    if (n) {
      const r = this.telInput.getAttribute("name") || "", o = n(r);
      if (o.phone) {
        const a = (i = this.telInput.form) == null ? void 0 : i.querySelector(
          `input[name="${o.phone}"]`
        );
        a ? this.hiddenInput = a : (this.hiddenInput = A("input", {
          type: "hidden",
          name: o.phone
        }), t.appendChild(this.hiddenInput));
      }
      if (o.country) {
        const a = (s = this.telInput.form) == null ? void 0 : s.querySelector(
          `input[name="${o.country}"]`
        );
        a ? this.hiddenInputCountry = a : (this.hiddenInputCountry = A("input", {
          type: "hidden",
          name: o.country
        }), t.appendChild(this.hiddenInputCountry));
      }
    }
  }
  //* For each country: add a country list item <li> to the countryList <ul> container.
  _appendListItems() {
    const t = document.createDocumentFragment();
    for (let n = 0; n < this.countries.length; n++) {
      const i = this.countries[n], s = ce({
        [I.COUNTRY_ITEM]: !0
      }), r = A("li", {
        id: `iti-${this.id}__item-${i.iso2}`,
        class: s,
        tabindex: "-1",
        role: "option",
        [N.SELECTED]: "false"
      });
      r.dataset.dialCode = i.dialCode, r.dataset.countryCode = i.iso2, i.nodeById[this.id] = r, this.options.showFlags && A("div", { class: `${I.FLAG} iti__${i.iso2}` }, r);
      const o = A("span", { class: "iti__country-name" }, r);
      o.textContent = `${i.name} `;
      const a = A("span", { class: "iti__dial-code" }, o);
      this.isRTL && a.setAttribute("dir", "ltr"), a.textContent = `(+${i.dialCode})`, t.appendChild(r);
    }
    this.countryList.appendChild(t);
  }
  //* Update the input padding to make space for the selected country/dial code.
  updateInputPadding() {
    if (this.selectedCountry) {
      const t = this.options.separateDialCode ? it.SANE_SELECTED_WITH_DIAL_WIDTH : it.SANE_SELECTED_NO_DIAL_WIDTH, i = (this.selectedCountry.offsetWidth || this._getHiddenSelectedCountryWidth() || t) + it.INPUT_PADDING_EXTRA_LEFT;
      this.telInput.style.paddingLeft = `${i}px`;
    }
  }
  static getBody() {
    let t;
    try {
      t = window.top.document.body;
    } catch {
      t = document.body;
    }
    return t;
  }
  //* When input is in a hidden container during init, we cannot calculate the selected country width.
  //* Fix: clone the markup, make it invisible, add it to the end of the DOM, and then measure it's width.
  //* To get the right styling to apply, all we need is a shallow clone of the container,
  //* and then to inject a deep clone of the selectedCountry element.
  _getHiddenSelectedCountryWidth() {
    if (this.telInput.parentNode) {
      const t = Vt.getBody(), n = this.telInput.parentNode.cloneNode(
        !1
      );
      n.style.visibility = "hidden", t.appendChild(n);
      const i = this.countryContainer.cloneNode();
      n.appendChild(i);
      const s = this.selectedCountry.cloneNode(
        !0
      );
      i.appendChild(s);
      const r = s.offsetWidth;
      return t.removeChild(n), r;
    }
    return 0;
  }
  // this is run before we add the dropdown to the DOM
  getHiddenInlineDropdownHeight() {
    const t = Vt.getBody();
    this.dropdownContent.classList.remove(I.HIDE);
    const n = A("div", { class: "iti iti--inline-dropdown" });
    n.appendChild(this.dropdownContent), n.style.visibility = "hidden", t.appendChild(n);
    const i = this.dropdownContent.offsetHeight;
    return t.removeChild(n), n.style.visibility = "", this.dropdownContent.classList.add(I.HIDE), i > 0 ? i : it.SANE_DROPDOWN_HEIGHT;
  }
  //* Update search results text (for a11y).
  updateSearchResultsA11yText() {
    const { i18n: t } = this.options, n = this.countryList.childElementCount;
    this.searchResultsA11yText.textContent = t.searchSummaryAria(n);
  }
  //* Country search: Filter the countries according to the search query.
  filterCountriesByQuery(t) {
    let n;
    t === "" ? n = this.countries : n = Ys(this.countries, t), this.filterCountries(n);
  }
  // Search input handlers
  doFilter() {
    const t = this.searchInput.value.trim();
    this.filterCountriesByQuery(t), this.searchInput.value ? this.searchClearButton.classList.remove(I.HIDE) : this.searchClearButton.classList.add(I.HIDE);
  }
  handleSearchChange() {
    this.searchKeyupTimer && clearTimeout(this.searchKeyupTimer), this.searchKeyupTimer = setTimeout(() => {
      this.doFilter(), this.searchKeyupTimer = null;
    }, Vn.SEARCH_DEBOUNCE_MS);
  }
  handleSearchClear() {
    this.searchInput.value = "", this.searchInput.focus(), this.doFilter();
  }
  //* Check if an element is visible within it's container, else scroll until it is.
  scrollTo(t) {
    const n = this.countryList, i = document.documentElement.scrollTop, s = n.offsetHeight, r = n.getBoundingClientRect().top + i, o = r + s, a = t.offsetHeight, l = t.getBoundingClientRect().top + i, u = l + a, c = l - r + n.scrollTop;
    if (l < r)
      n.scrollTop = c;
    else if (u > o) {
      const d = s - a;
      n.scrollTop = c - d;
    }
  }
  //* Remove highlighting from the previous list item and highlight the new one.
  highlightListItem(t, n) {
    const i = this.highlightedItem;
    if (i && i.classList.remove(I.HIGHLIGHT), this.highlightedItem = t, this.highlightedItem && (this.highlightedItem.classList.add(I.HIGHLIGHT), this.options.countrySearch)) {
      const s = this.highlightedItem.getAttribute("id") || "";
      this.searchInput.setAttribute(N.ACTIVE_DESCENDANT, s);
    }
    n && this.highlightedItem.focus();
  }
  //* Highlight the next/prev item in the list (and ensure it is visible).
  handleUpDownKey(t) {
    var i, s;
    let n = t === M.ARROW_UP ? (i = this.highlightedItem) == null ? void 0 : i.previousElementSibling : (s = this.highlightedItem) == null ? void 0 : s.nextElementSibling;
    !n && this.countryList.childElementCount > 1 && (n = t === M.ARROW_UP ? this.countryList.lastElementChild : this.countryList.firstElementChild), n && (this.scrollTo(n), this.highlightListItem(n, !1));
  }
  // Update the selected list item in the dropdown
  updateSelectedItem(t) {
    var n;
    if (this.selectedItem && this.selectedItem.dataset.countryCode !== t && (this.selectedItem.setAttribute(N.SELECTED, "false"), (n = this.selectedItem.querySelector(".iti__country-check")) == null || n.remove(), this.selectedItem = null), t && !this.selectedItem) {
      const i = this.countryList.querySelector(
        `[data-country-code="${t}"]`
      );
      if (i) {
        i.setAttribute(N.SELECTED, "true");
        const s = A(
          "span",
          { class: "iti__country-check", [N.HIDDEN]: "true" },
          i
        );
        s.innerHTML = Xs(), this.selectedItem = i;
      }
    }
  }
  //* Country search: Filter the country list to the given array of countries.
  filterCountries(t) {
    this.countryList.innerHTML = "";
    let n = !0;
    for (const i of t) {
      const s = i.nodeById[this.id];
      s && (this.countryList.appendChild(s), n && (this.highlightListItem(s, !1), n = !1));
    }
    n ? (this.highlightListItem(null, !1), this.searchNoResults && this.searchNoResults.classList.remove(I.HIDE)) : this.searchNoResults && this.searchNoResults.classList.add(I.HIDE), this.countryList.scrollTop = 0, this.updateSearchResultsA11yText();
  }
  destroy() {
    this.telInput.iti = void 0, delete this.telInput.dataset.intlTelInputId, this.options.separateDialCode && (this.telInput.style.paddingLeft = this.originalPaddingLeft);
    const t = this.telInput.parentNode;
    t.before(this.telInput), t.remove(), this.telInput = null, this.countryContainer = null, this.selectedCountry = null, this.selectedCountryInner = null, this.selectedDialCode = null, this.dropdownArrow = null, this.dropdownContent = null, this.searchInput = null, this.searchIcon = null, this.searchClearButton = null, this.searchNoResults = null, this.searchResultsA11yText = null, this.countryList = null, this.dropdownForContainer = null, this.hiddenInput = null, this.hiddenInputCountry = null, this.highlightedItem = null, this.selectedItem = null;
    for (const n of this.countries)
      delete n.nodeById[this.id];
    this.countries = null;
  }
  // UI: Open the dropdown (DOM only).
  openDropdown() {
    const {
      countrySearch: t,
      dropdownAlwaysOpen: n,
      dropdownContainer: i
    } = this.options;
    if (this._maybeEnsureDropdownWidthSet(), i)
      this._handleDropdownContainer();
    else {
      const s = this._shouldPositionInlineDropdownBelowInput(), r = this.telInput.offsetHeight + it.DROPDOWN_MARGIN;
      s ? this.dropdownContent.style.top = `${r}px` : this.dropdownContent.style.bottom = `${r}px`;
    }
    if (this.dropdownContent.classList.remove(I.HIDE), this.selectedCountry.setAttribute(N.EXPANDED, "true"), t) {
      const s = this.countryList.firstElementChild;
      s && (this.highlightListItem(s, !1), this.countryList.scrollTop = 0), n || this.searchInput.focus();
    }
    this.dropdownArrow.classList.add(I.ARROW_UP);
  }
  // UI: Close the dropdown (DOM only).
  closeDropdown() {
    const { countrySearch: t, dropdownContainer: n } = this.options;
    this.dropdownContent.classList.add(I.HIDE), this.selectedCountry.setAttribute(N.EXPANDED, "false"), t && (this.searchInput.removeAttribute(N.ACTIVE_DESCENDANT), this.highlightedItem && (this.highlightedItem.classList.remove(I.HIGHLIGHT), this.highlightedItem = null)), this.dropdownArrow.classList.remove(I.ARROW_UP), n ? (this.dropdownForContainer.remove(), this.dropdownForContainer.style.top = "", this.dropdownForContainer.style.bottom = "") : (this.dropdownContent.style.top = "", this.dropdownContent.style.bottom = "");
  }
  _shouldPositionInlineDropdownBelowInput() {
    if (this.options.dropdownAlwaysOpen)
      return !0;
    const t = this.telInput.getBoundingClientRect(), n = t.top, i = window.innerHeight - t.bottom;
    return i >= this.inlineDropdownHeight || i >= n;
  }
  // inject dropdown into container and apply positioning styles
  _handleDropdownContainer() {
    const { dropdownContainer: t, useFullscreenPopup: n } = this.options;
    if (t) {
      if (!n) {
        const i = this.telInput.getBoundingClientRect();
        this.dropdownForContainer.style.left = `${i.left}px`, this._shouldPositionInlineDropdownBelowInput() ? this.dropdownForContainer.style.top = `${i.bottom + it.DROPDOWN_MARGIN}px` : (this.dropdownForContainer.style.top = "unset", this.dropdownForContainer.style.bottom = `${window.innerHeight - i.top + it.DROPDOWN_MARGIN}px`);
      }
      t.appendChild(this.dropdownForContainer);
    }
  }
  // UI: Whether the dropdown is currently closed (hidden).
  isDropdownClosed() {
    return this.dropdownContent.classList.contains(I.HIDE);
  }
  setCountry(t) {
    const { allowDropdown: n, showFlags: i, separateDialCode: s, i18n: r } = this.options, { name: o, dialCode: a, iso2: l = "" } = t;
    if (n && this.updateSelectedItem(l), this.selectedCountry) {
      const u = l && i ? `${I.FLAG} iti__${l}` : `${I.FLAG} ${I.GLOBE}`;
      let c, d, f;
      l ? (d = o, c = r.selectedCountryAriaLabel.replace("${countryName}", o).replace("${dialCode}", `+${a}`), f = i ? "" : tn()) : (d = r.noCountrySelected, c = r.noCountrySelected, f = tn()), this.selectedCountryInner.className = u, this.selectedCountry.setAttribute("title", d), this.selectedCountry.setAttribute(N.LABEL, c), this.selectedCountryInner.innerHTML = f;
    }
    if (s) {
      const u = a ? `+${a}` : "";
      this.selectedDialCode.textContent = u, this.updateInputPadding();
    }
  }
}
const Qs = (e) => {
  const { onlyCountries: t, excludeCountries: n } = e;
  if (t != null && t.length) {
    const i = t.map(
      (s) => s.toLowerCase()
    );
    return _t.filter(
      (s) => i.includes(s.iso2)
    );
  } else if (n != null && n.length) {
    const i = n.map(
      (s) => s.toLowerCase()
    );
    return _t.filter(
      (s) => !i.includes(s.iso2)
    );
  }
  return _t;
}, Zs = (e, t) => {
  const { countryNameLocale: n, i18n: i } = t;
  let s;
  try {
    typeof Intl < "u" && typeof Intl.DisplayNames == "function" ? s = new Intl.DisplayNames(n, {
      type: "region"
    }) : s = null;
  } catch (r) {
    console.error(r), s = null;
  }
  for (const r of e)
    r.name = i[r.iso2] || (s == null ? void 0 : s.of(r.iso2.toUpperCase())) || "";
}, Js = (e) => {
  const t = /* @__PURE__ */ new Set();
  let n = 0;
  const i = {}, s = (o, a) => {
    if (!o || !a)
      return;
    a.length > n && (n = a.length), i.hasOwnProperty(a) || (i[a] = []);
    const l = i[a];
    l.includes(o) || l.push(o);
  }, r = [...e].sort((o, a) => o.priority - a.priority);
  for (const o of r) {
    t.has(o.dialCode) || t.add(o.dialCode);
    for (let a = 1; a < o.dialCode.length; a++) {
      const l = o.dialCode.substring(0, a);
      s(o.iso2, l);
    }
    if (s(o.iso2, o.dialCode), o.areaCodes) {
      const a = i[o.dialCode][0];
      for (const l of o.areaCodes) {
        for (let u = 1; u < l.length; u++) {
          const c = l.substring(0, u), d = o.dialCode + c;
          s(a, d), s(o.iso2, d);
        }
        s(o.iso2, o.dialCode + l);
      }
    }
  }
  return { dialCodes: t, dialCodeMaxLen: n, dialCodeToIso2Map: i };
}, tr = (e, t) => {
  t.countryOrder && (t.countryOrder = t.countryOrder.map(
    (n) => n.toLowerCase()
  )), e.sort((n, i) => {
    const { countryOrder: s } = t;
    if (s) {
      const r = s.indexOf(n.iso2), o = s.indexOf(i.iso2), a = r > -1, l = o > -1;
      if (a || l)
        return a && l ? r - o : a ? -1 : 1;
    }
    return n.name.localeCompare(i.name);
  });
}, er = (e) => {
  for (const t of e)
    t.normalisedName = Kn(t.name), t.initials = t.normalisedName.split(/[^a-z]/).map((n) => n[0]).join(""), t.dialCodePlus = `+${t.dialCode}`;
}, nr = /* @__PURE__ */ new Set([
  "800",
  "808",
  "870",
  "881",
  "882",
  "883",
  "888",
  "979"
]), en = (e) => {
  const t = xt(e).slice(0, 3);
  return e.startsWith("+") && nr.has(t);
}, nn = (e, t, n, i) => {
  let s = e;
  if (n && t) {
    const r = `+${i.dialCode}`, o = s[r.length] === " " || s[r.length] === "-" ? r.length + 1 : r.length;
    s = s.substring(o);
  }
  return s;
}, ir = (e, t, n, i, s) => {
  const r = n ? n.formatNumberAsYouType(e, i.iso2) : e, { dialCode: o } = i;
  return s && t.charAt(0) !== "+" && r.includes(`+${o}`) ? (r.split(`+${o}`)[1] || "").trim() : r;
}, sr = (e, t, n, i) => {
  if (n === 0 && !i)
    return 0;
  let s = 0;
  for (let r = 0; r < t.length; r++) {
    if (/[+0-9]/.test(t[r]) && s++, s === e && !i)
      return r + 1;
    if (i && s === e + 1)
      return r;
  }
  return t.length;
}, rr = [
  "800",
  "822",
  "833",
  "844",
  "855",
  "866",
  "877",
  "880",
  "881",
  "882",
  "883",
  "884",
  "885",
  "886",
  "887",
  "888",
  "889"
], sn = (e) => {
  const t = xt(e);
  if (t.startsWith(jn.NANP) && t.length >= 4) {
    const n = t.substring(1, 4);
    return rr.includes(n);
  }
  return !1;
};
class or {
  constructor() {
  }
  // If any Arabic-Indic digits, then label it as that set. Same for Persian. Otherwise assume ASCII.
  updateNumeralSet(t) {
    /[\u0660-\u0669]/.test(t) ? this.userNumeralSet = "arabic-indic" : /[\u06F0-\u06F9]/.test(t) ? this.userNumeralSet = "persian" : this.userNumeralSet = "ascii";
  }
  // Denormalise ASCII 0-9 to the user's numeral set, if known. If not known, return the string as-is.
  denormalise(t, n) {
    if (this.userNumeralSet || this.updateNumeralSet(n), this.userNumeralSet === "ascii")
      return t;
    const i = this.userNumeralSet === "arabic-indic" ? 1632 : 1776;
    return t.replace(/[0-9]/g, (s) => String.fromCharCode(i + Number(s)));
  }
  // Normalize Eastern Arabic (U+0660-0669) and Persian/Extended Arabic-Indic (U+06F0-06F9) numerals to ASCII 0-9
  normalise(t) {
    if (!t)
      return "";
    if (this.updateNumeralSet(t), this.userNumeralSet === "ascii")
      return t;
    const n = this.userNumeralSet === "arabic-indic" ? 1632 : 1776, i = this.userNumeralSet === "arabic-indic" ? /[\u0660-\u0669]/g : /[\u06F0-\u06F9]/g;
    return t.replace(i, (s) => String.fromCharCode(48 + (s.charCodeAt(0) - n)));
  }
  isAscii() {
    return this.userNumeralSet === "ascii";
  }
}
let lr = 0;
const ar = new Set(_t.map((e) => e.iso2)), de = (e) => ar.has(e);
class ur {
  constructor(t, n = {}) {
    this.id = lr++, Vt.validateInput(t);
    const i = js(n);
    this.options = { ...Oe, ...i }, Gs(this.options), this.ui = new Vt(t, this.options, this.id), this.isAndroid = $s(), this.numerals = new or(), this.promise = this._createInitPromises(this.options), this.countries = Qs(this.options);
    const { dialCodes: s, dialCodeMaxLen: r, dialCodeToIso2Map: o } = Js(
      this.countries
    );
    this.dialCodes = s, this.dialCodeMaxLen = r, this.dialCodeToIso2Map = o, this.countryByIso2 = new Map(this.countries.map((a) => [a.iso2, a])), this._init();
  }
  _getTelInputValue() {
    const t = this.ui.telInput.value.trim();
    return this.numerals.normalise(t);
  }
  _setTelInputValue(t) {
    const n = this.ui.telInput.value;
    this.ui.telInput.value = this.numerals.denormalise(t, n);
  }
  _createInitPromises(t) {
    const { initialCountry: n, geoIpLookup: i, loadUtils: s } = t, r = n === Ft.AUTO && !!i, o = !!s && !C.utils;
    let a;
    r ? a = new Promise((u, c) => {
      this.resolveAutoCountryPromise = u, this.rejectAutoCountryPromise = c;
    }) : (a = Promise.resolve(void 0), this.resolveAutoCountryPromise = () => {
    }, this.rejectAutoCountryPromise = () => {
    });
    let l;
    return o ? l = new Promise((u, c) => {
      this.resolveUtilsScriptPromise = u, this.rejectUtilsScriptPromise = c;
    }) : (l = Promise.resolve(void 0), this.resolveUtilsScriptPromise = () => {
    }, this.rejectUtilsScriptPromise = () => {
    }), Promise.all([a, l]);
  }
  //* Can't be private as it's called from intlTelInput convenience wrapper.
  _init() {
    this.selectedCountryData = {}, this.abortController = new AbortController(), this._processCountryData(), this.ui.generateMarkup(this.countries), this._setInitialState(), this._initListeners(), this._initRequests(), this.options.dropdownAlwaysOpen && this._openDropdown();
  }
  //********************
  //*  PRIVATE METHODS
  //********************
  //* Prepare all of the country data, including onlyCountries, excludeCountries, countryOrder options.
  _processCountryData() {
    Zs(this.countries, this.options), tr(this.countries, this.options), er(this.countries);
  }
  //* Set the initial state of the input value and the selected country by:
  //* 1. Extracting a dial code from the given number
  //* 2. Using explicit initialCountry
  _setInitialState(t = !1) {
    const n = this.ui.telInput.getAttribute("value"), i = this.numerals.normalise(n), s = this._getTelInputValue(), o = i && i.startsWith("+") && (!s || !s.startsWith("+")) ? i : s, a = this._getDialCode(o), l = sn(o), { initialCountry: u, geoIpLookup: c } = this.options, f = u === Ft.AUTO && c && !t, p = u.toLowerCase(), _ = de(p);
    a ? l ? _ ? this._setCountry(p) : f || this._setCountry(Ms.ISO2) : this._updateCountryFromNumber(o) : _ ? this._setCountry(p) : f || this._setCountry(""), o && this._updateValFromNumber(o);
  }
  //* Initialise the main event listeners: input keyup, and click selected country.
  _initListeners() {
    this._initTelInputListeners(), this.options.allowDropdown && this._initDropdownListeners(), (this.ui.hiddenInput || this.ui.hiddenInputCountry) && this.ui.telInput.form && this._initHiddenInputListener();
  }
  //* Update hidden input on form submit.
  _initHiddenInputListener() {
    var n;
    const t = () => {
      this.ui.hiddenInput && (this.ui.hiddenInput.value = this.getNumber()), this.ui.hiddenInputCountry && (this.ui.hiddenInputCountry.value = this.selectedCountryData.iso2 || "");
    };
    (n = this.ui.telInput.form) == null || n.addEventListener("submit", t, {
      signal: this.abortController.signal
    });
  }
  //* initialise the dropdown listeners.
  _initDropdownListeners() {
    const t = this.abortController.signal, n = (o) => {
      this.ui.isDropdownClosed() ? this.ui.telInput.focus() : o.preventDefault();
    }, i = this.ui.telInput.closest("label");
    i && i.addEventListener("click", n, { signal: t });
    const s = () => {
      this.ui.isDropdownClosed() && !this.ui.telInput.disabled && !this.ui.telInput.readOnly && this._openDropdown();
    };
    this.ui.selectedCountry.addEventListener(
      "click",
      s,
      {
        signal: t
      }
    );
    const r = (o) => {
      const a = [M.ARROW_UP, M.ARROW_DOWN, M.SPACE, M.ENTER];
      this.ui.isDropdownClosed() && a.includes(o.key) && (o.preventDefault(), o.stopPropagation(), this._openDropdown()), o.key === M.TAB && this._closeDropdown();
    };
    this.ui.countryContainer.addEventListener(
      "keydown",
      r,
      { signal: t }
    );
  }
  //* Init requests: utils script / geo ip lookup.
  _initRequests() {
    const { loadUtils: t, initialCountry: n, geoIpLookup: i } = this.options;
    if (t && !C.utils) {
      const r = () => {
        var o;
        (o = C.attachUtils(t)) == null || o.catch(() => {
        });
      };
      if (C.documentReady())
        r();
      else {
        const o = () => {
          r();
        };
        window.addEventListener("load", o, {
          signal: this.abortController.signal
        });
      }
    } else
      this.resolveUtilsScriptPromise();
    n === Ft.AUTO && i && (this.selectedCountryData.iso2 ? this.resolveAutoCountryPromise() : this._loadAutoCountry());
  }
  //* Perform the geo ip lookup.
  _loadAutoCountry() {
    if (C.autoCountry)
      this.handleAutoCountry();
    else if (this.ui.selectedCountryInner.classList.add(I.LOADING), !C.startedLoadingAutoCountry && (C.startedLoadingAutoCountry = !0, typeof this.options.geoIpLookup == "function")) {
      const t = (i = "") => {
        this.ui.selectedCountryInner.classList.remove(I.LOADING);
        const s = i.toLowerCase();
        de(s) ? (C.autoCountry = s, setTimeout(() => Ht("handleAutoCountry"))) : Ht("handleAutoCountryFailure");
      }, n = () => {
        this.ui.selectedCountryInner.classList.remove(I.LOADING), Ht("handleAutoCountryFailure");
      };
      this.options.geoIpLookup(t, n);
    }
  }
  _openDropdownWithPlus() {
    this._openDropdown(), this.ui.searchInput.value = "+", this.ui.filterCountriesByQuery("");
  }
  //* Initialize the tel input listeners.
  _initTelInputListeners() {
    this._bindInputListener(), this._maybeBindKeydownListener(), this._maybeBindPasteListener();
  }
  _bindInputListener() {
    const {
      strictMode: t,
      formatAsYouType: n,
      separateDialCode: i,
      allowDropdown: s,
      countrySearch: r
    } = this.options;
    let o = !1;
    ct.ALPHA_UNICODE.test(this._getTelInputValue()) && (o = !0);
    const a = (l) => {
      const u = this._getTelInputValue();
      if (this.isAndroid && (l == null ? void 0 : l.data) === "+" && i && s && r) {
        const _ = this.ui.telInput.selectionStart || 0, m = u.substring(0, _ - 1), v = u.substring(_);
        this._setTelInputValue(m + v), this._openDropdownWithPlus();
        return;
      }
      this._updateCountryFromNumber(u) && this._triggerCountryChange();
      const c = (l == null ? void 0 : l.data) && ct.NON_PLUS_NUMERIC.test(l.data), d = (l == null ? void 0 : l.inputType) === ze.PASTE && u;
      c || d && !t ? o = !0 : ct.NON_PLUS_NUMERIC.test(u) || (o = !1);
      const f = (l == null ? void 0 : l.detail) && l.detail.isSetNumber, p = this.numerals.isAscii();
      if (n && !o && !f && p) {
        const _ = this.ui.telInput.selectionStart || 0, v = u.substring(
          0,
          _
        ).replace(
          ct.NON_PLUS_NUMERIC_GLOBAL,
          ""
        ).length, y = (l == null ? void 0 : l.inputType) === ze.DELETE_FWD, b = this._getFullNumber(), h = ir(
          b,
          u,
          C.utils,
          this.selectedCountryData,
          i
        ), g = sr(
          v,
          h,
          _,
          y
        );
        this._setTelInputValue(h), this.ui.telInput.setSelectionRange(g, g);
      }
      if (i && u.startsWith("+") && this.selectedCountryData.dialCode) {
        const _ = nn(u, !0, i, this.selectedCountryData);
        this._setTelInputValue(_);
      }
    };
    this.ui.telInput.addEventListener(
      "input",
      a,
      {
        signal: this.abortController.signal
      }
    );
  }
  _maybeBindKeydownListener() {
    const { strictMode: t, separateDialCode: n, allowDropdown: i, countrySearch: s } = this.options;
    if (t || n) {
      const r = (o) => {
        if (o.key && o.key.length === 1 && !o.altKey && !o.ctrlKey && !o.metaKey) {
          if (n && i && s && o.key === "+") {
            o.preventDefault(), this._openDropdownWithPlus();
            return;
          }
          if (t) {
            const a = this._getTelInputValue(), u = !a.startsWith("+") && this.ui.telInput.selectionStart === 0 && o.key === "+", c = this.numerals.normalise(o.key), d = /^[0-9]$/.test(c), f = n ? d : u || d, p = this.ui.telInput, _ = p.selectionStart, m = p.selectionEnd, v = a.slice(0, _), y = a.slice(m), b = v + o.key + y, h = this._getFullNumber(b), g = C.utils.getCoreNumber(
              h,
              this.selectedCountryData.iso2
            ), L = this.maxCoreNumberLength && g.length > this.maxCoreNumberLength, O = this._getNewCountryFromNumber(h) !== null;
            (!f || L && !O && !u) && o.preventDefault();
          }
        }
      };
      this.ui.telInput.addEventListener("keydown", r, {
        signal: this.abortController.signal
      });
    }
  }
  _maybeBindPasteListener() {
    if (this.options.strictMode) {
      const t = (n) => {
        n.preventDefault();
        const i = this.ui.telInput, s = i.selectionStart, r = i.selectionEnd, o = this._getTelInputValue(), a = o.slice(0, s), l = o.slice(r), u = this.selectedCountryData.iso2, c = n.clipboardData.getData("text"), d = this.numerals.normalise(c), f = s === 0 && r > 0, p = !o.startsWith("+") || f, _ = d.replace(ct.NON_PLUS_NUMERIC_GLOBAL, ""), m = _.startsWith("+"), v = _.replace(/\+/g, ""), y = m && p ? `+${v}` : v;
        let b = a + y + l;
        if (b.length > 5) {
          let g = C.utils.getCoreNumber(b, u);
          for (; g.length === 0 && b.length > 0; )
            b = b.slice(0, -1), g = C.utils.getCoreNumber(b, u);
          if (!g)
            return;
          if (this.maxCoreNumberLength && g.length > this.maxCoreNumberLength)
            if (i.selectionEnd === o.length) {
              const L = g.length - this.maxCoreNumberLength;
              b = b.slice(0, b.length - L);
            } else
              return;
        }
        this._setTelInputValue(b);
        const h = s + y.length;
        i.setSelectionRange(h, h), i.dispatchEvent(new InputEvent("input", { bubbles: !0 }));
      };
      this.ui.telInput.addEventListener("paste", t, {
        signal: this.abortController.signal
      });
    }
  }
  //* Adhere to the input's maxlength attr.
  _cap(t) {
    const n = Number(this.ui.telInput.getAttribute("maxlength"));
    return n && t.length > n ? t.substring(0, n) : t;
  }
  //* Trigger a custom event on the input (typed via ItiEventMap).
  _trigger(t, n = {}) {
    const i = new CustomEvent(t, {
      bubbles: !0,
      cancelable: !0,
      detail: n
    });
    this.ui.telInput.dispatchEvent(i);
  }
  //* Open the dropdown.
  _openDropdown() {
    const { dropdownContainer: t, useFullscreenPopup: n } = this.options;
    if (this.dropdownAbortController = new AbortController(), this.ui.openDropdown(), !n && t) {
      const i = () => this._closeDropdown();
      window.addEventListener("scroll", i, {
        signal: this.dropdownAbortController.signal
      });
    }
    this._bindDropdownListeners(), this._trigger(Kt.OPEN_COUNTRY_DROPDOWN);
  }
  //* We only bind dropdown listeners when the dropdown is open.
  _bindDropdownListeners() {
    const t = this.dropdownAbortController.signal;
    this._bindDropdownMouseoverListener(t), this._bindDropdownCountryClickListener(t), this.options.dropdownAlwaysOpen || this._bindDropdownClickOffListener(t), this._bindDropdownKeydownListener(t), this.options.countrySearch && this._bindDropdownSearchListeners(t);
  }
  //* When mouse over a list item, just highlight that one
  //* we add the class "highlight", so if they hit "enter" we know which one to select.
  _bindDropdownMouseoverListener(t) {
    const n = (i) => {
      var r;
      const s = (r = i.target) == null ? void 0 : r.closest(
        `.${I.COUNTRY_ITEM}`
      );
      s && this.ui.highlightListItem(s, !1);
    };
    this.ui.countryList.addEventListener(
      "mouseover",
      n,
      {
        signal: t
      }
    );
  }
  //* Listen for country selection.
  _bindDropdownCountryClickListener(t) {
    const n = (i) => {
      var r;
      const s = (r = i.target) == null ? void 0 : r.closest(
        `.${I.COUNTRY_ITEM}`
      );
      s && this._selectListItem(s);
    };
    this.ui.countryList.addEventListener("click", n, {
      signal: t
    });
  }
  //* Click off to close (except when this initial opening click is bubbling up).
  //* We cannot just stopPropagation as it may be needed to close another instance.
  _bindDropdownClickOffListener(t) {
    const n = (i) => {
      !!i.target.closest(
        `#iti-${this.id}__dropdown-content`
      ) || this._closeDropdown();
    };
    setTimeout(() => {
      document.documentElement.addEventListener(
        "click",
        n,
        { signal: t }
      );
    }, 0);
  }
  //* Listen for up/down scrolling, enter to select, or escape to close.
  //* Use keydown as keypress doesn't fire for non-char keys and we want to catch if they
  //* just hit down and hold it to scroll down (no keyup event).
  //* Listen on the document because that's where key events are triggered if no input has focus.
  _bindDropdownKeydownListener(t) {
    let n = "", i = null;
    const s = (r) => {
      [
        M.ARROW_UP,
        M.ARROW_DOWN,
        M.ENTER,
        M.ESC
      ].includes(r.key) && (r.preventDefault(), r.stopPropagation(), r.key === M.ARROW_UP || r.key === M.ARROW_DOWN ? this.ui.handleUpDownKey(r.key) : r.key === M.ENTER ? this._handleEnterKey() : r.key === M.ESC && (this._closeDropdown(), this.ui.selectedCountry.focus())), !this.options.countrySearch && ct.HIDDEN_SEARCH_CHAR.test(r.key) && (r.stopPropagation(), i && clearTimeout(i), n += r.key.toLowerCase(), this._searchForCountry(n), i = setTimeout(() => {
        n = "";
      }, Vn.HIDDEN_SEARCH_RESET_MS));
    };
    document.addEventListener("keydown", s, { signal: t });
  }
  //* Search input listeners when countrySearch enabled.
  _bindDropdownSearchListeners(t) {
    this.ui.searchInput.addEventListener(
      "input",
      () => this.ui.handleSearchChange(),
      { signal: t }
    ), this.ui.searchClearButton.addEventListener(
      "click",
      () => this.ui.handleSearchClear(),
      { signal: t }
    );
  }
  //* Hidden search (countrySearch disabled): Find the first list item whose name starts with the query string.
  _searchForCountry(t) {
    const n = Ks(this.countries, t);
    if (n) {
      const i = n.nodeById[this.id];
      this.ui.highlightListItem(i, !1), this.ui.scrollTo(i);
    }
  }
  //* Select the currently highlighted item.
  _handleEnterKey() {
    this.ui.highlightedItem && this._selectListItem(this.ui.highlightedItem);
  }
  //* Update the input's value to the given val (format first if possible)
  //* NOTE: this is called from _setInitialState, handleUtils and setNumber.
  _updateValFromNumber(t) {
    const { formatOnDisplay: n, nationalMode: i, separateDialCode: s } = this.options;
    let r = t;
    if (n && C.utils && this.selectedCountryData) {
      const o = en(t), a = i && !o || !r.startsWith("+") && !s, { NATIONAL: l, INTERNATIONAL: u } = C.utils.numberFormat, c = a ? l : u;
      r = C.utils.formatNumber(
        r,
        this.selectedCountryData.iso2,
        c
      );
    }
    r = this._beforeSetNumber(r), this._setTelInputValue(r);
  }
  //* Check if need to select a new country based on the given number
  //* Note: called from _setInitialState, keyup handler, setNumber.
  _updateCountryFromNumber(t) {
    const n = this._getNewCountryFromNumber(t);
    return n !== null ? this._setCountry(n) : !1;
  }
  // if there is a selected country, and the number doesn't start with a dial code, then add it
  _ensureHasDialCode(t) {
    const { dialCode: n, nationalPrefix: i } = this.selectedCountryData;
    if (t.startsWith("+") || !n)
      return t;
    const o = i && t.startsWith(i) && !this.options.separateDialCode ? t.substring(1) : t;
    return `+${n}${o}`;
  }
  //* Get the new country based on the input number, or return null if no change, or empty string if should be empty (e.g. if they type an invalid dial code).
  _getNewCountryFromNumber(t) {
    const n = t.indexOf("+");
    let i = n ? t.substring(n) : t;
    const s = this.selectedCountryData.iso2, r = this.selectedCountryData.dialCode;
    i = this._ensureHasDialCode(i);
    const o = this._getDialCode(i, !0), a = xt(i);
    if (o) {
      const l = xt(o), u = this.dialCodeToIso2Map[l];
      if (u.length === 1)
        return u[0] === s ? null : u[0];
      if (!s && this.defaultCountry && u.includes(this.defaultCountry))
        return this.defaultCountry;
      if (r === jn.NANP && sn(a))
        return null;
      const { areaCodes: d, priority: f } = this.selectedCountryData;
      if (d) {
        const y = d.map(
          (b) => `${r}${b}`
        );
        for (const b of y)
          if (a.startsWith(b))
            return null;
      }
      const _ = d && !(f === 0) && a.length > l.length, m = s && u.includes(s) && !_, v = s === u[0];
      if (!m && !v)
        return u[0];
    } else if (i.startsWith("+") && a.length) {
      const l = this.selectedCountryData.dialCode || "";
      return l && l.startsWith(a) ? null : "";
    } else if ((!i || i === "+") && !s && this.defaultCountry)
      return this.defaultCountry;
    return null;
  }
  //* Update the selected country, dial code (if separateDialCode), placeholder, title, and selected list item.
  //* Note: called from _setInitialState, _updateCountryFromNumber, _selectListItem, setCountry.
  _setCountry(t) {
    const n = this.selectedCountryData.iso2 || "";
    return this.selectedCountryData = t ? this.countryByIso2.get(t) : {}, this.selectedCountryData.iso2 && (this.defaultCountry = this.selectedCountryData.iso2), this.ui.setCountry(this.selectedCountryData), this._updatePlaceholder(), this._updateMaxLength(), n !== t;
  }
  //* Update the maximum valid number length for the currently selected country.
  _updateMaxLength() {
    const { strictMode: t, placeholderNumberType: n, allowedNumberTypes: i } = this.options, { iso2: s } = this.selectedCountryData;
    if (t && C.utils)
      if (s) {
        const r = C.utils.numberType[n];
        let o = C.utils.getExampleNumber(
          s,
          !1,
          r,
          !0
        ), a = o;
        for (; C.utils.isPossibleNumber(
          o,
          s,
          i
        ); )
          a = o, o += "0";
        const l = C.utils.getCoreNumber(a, s);
        this.maxCoreNumberLength = l.length, s === "by" && (this.maxCoreNumberLength = l.length + 1);
      } else
        this.maxCoreNumberLength = null;
  }
  //* Update the input placeholder to an example number from the currently selected country.
  _updatePlaceholder() {
    const {
      autoPlaceholder: t,
      placeholderNumberType: n,
      nationalMode: i,
      customPlaceholder: s
    } = this.options, r = t === ne.AGGRESSIVE || !this.ui.hadInitialPlaceholder && t === ne.POLITE;
    if (C.utils && r) {
      const o = C.utils.numberType[n];
      let a = this.selectedCountryData.iso2 ? C.utils.getExampleNumber(
        this.selectedCountryData.iso2,
        i,
        o
      ) : "";
      a = this._beforeSetNumber(a), typeof s == "function" && (a = s(a, this.selectedCountryData)), this.ui.telInput.setAttribute("placeholder", a);
    }
  }
  //* Called when the user selects a list item from the dropdown.
  _selectListItem(t) {
    const n = t.dataset[Xe.COUNTRY_CODE], i = this._setCountry(n);
    this._closeDropdown();
    const s = t.dataset[Xe.DIAL_CODE];
    if (this._updateDialCode(s), this.options.formatOnDisplay) {
      const r = this._getTelInputValue();
      this._updateValFromNumber(r);
    }
    this.ui.telInput.focus(), i && this._triggerCountryChange();
  }
  //* Close the dropdown and unbind any listeners.
  _closeDropdown(t) {
    this.ui.isDropdownClosed() || this.options.dropdownAlwaysOpen && !t || (this.ui.closeDropdown(), this.dropdownAbortController.abort(), this.dropdownAbortController = null, this._trigger(Kt.CLOSE_COUNTRY_DROPDOWN));
  }
  //* Replace any existing dial code with the new one
  //* Note: called from _selectListItem and setCountry
  _updateDialCode(t) {
    const n = this._getTelInputValue(), i = `+${t}`;
    let s;
    if (n.startsWith("+")) {
      const r = this._getDialCode(n);
      r ? s = n.replace(r, i) : s = i, this._setTelInputValue(s);
    }
  }
  //* Try and extract a valid international dial code from a full telephone number.
  //* Note: returns the raw string inc plus character and any whitespace/dots etc.
  _getDialCode(t, n) {
    let i = "";
    if (t.startsWith("+")) {
      let s = "", r = !1;
      for (let o = 0; o < t.length; o++) {
        const a = t.charAt(o);
        if (/[0-9]/.test(a)) {
          if (s += a, !!!this.dialCodeToIso2Map[s])
            break;
          if (this.dialCodes.has(s)) {
            if (i = t.substring(0, o + 1), r = !0, !n)
              break;
          } else n && r && (i = t.substring(0, o + 1));
          if (s.length === this.dialCodeMaxLen)
            break;
        }
      }
    }
    return i;
  }
  //* Get the input val, adding the dial code if separateDialCode is enabled.
  _getFullNumber(t) {
    const n = t ? this.numerals.normalise(t) : this._getTelInputValue(), { dialCode: i } = this.selectedCountryData;
    let s;
    const r = xt(n);
    return this.options.separateDialCode && !n.startsWith("+") && i && r ? s = `+${i}` : s = "", s + n;
  }
  //* Remove the dial code if separateDialCode is enabled also cap the length if the input has a maxlength attribute
  _beforeSetNumber(t) {
    const n = !!this._getDialCode(t), i = nn(
      t,
      n,
      this.options.separateDialCode,
      this.selectedCountryData
    );
    return this._cap(i);
  }
  //* Trigger the 'countrychange' event.
  _triggerCountryChange() {
    this._trigger(Kt.COUNTRY_CHANGE);
  }
  //**************************
  //*  SECRET PUBLIC METHODS
  //**************************
  //* This is called when the geoip call returns.
  handleAutoCountry() {
    var t, n;
    if (!((t = this.ui) != null && t.telInput)) {
      (n = this.resolveAutoCountryPromise) == null || n.call(this);
      return;
    }
    this.options.initialCountry === Ft.AUTO && C.autoCountry && (this.defaultCountry = C.autoCountry, this.selectedCountryData.iso2 || this.ui.selectedCountryInner.classList.contains(I.GLOBE) || this.setCountry(this.defaultCountry), this.resolveAutoCountryPromise());
  }
  //* This is called when the geoip call fails or times out.
  handleAutoCountryFailure() {
    var t, n;
    if (!((t = this.ui) != null && t.telInput)) {
      (n = this.rejectAutoCountryPromise) == null || n.call(this);
      return;
    }
    this._setInitialState(!0), this.rejectAutoCountryPromise();
  }
  //* This is called when the utils request completes.
  handleUtils() {
    var t, n;
    if (!((t = this.ui) != null && t.telInput)) {
      (n = this.resolveUtilsScriptPromise) == null || n.call(this);
      return;
    }
    if (C.utils) {
      const i = this._getTelInputValue();
      i && this._updateValFromNumber(i), this.selectedCountryData.iso2 && (this._updatePlaceholder(), this._updateMaxLength());
    }
    this.resolveUtilsScriptPromise();
  }
  //* This is called when the utils request fails or times out.
  handleUtilsFailure(t) {
    var n, i;
    if (!((n = this.ui) != null && n.telInput)) {
      (i = this.rejectUtilsScriptPromise) == null || i.call(this, t);
      return;
    }
    this.rejectUtilsScriptPromise(t);
  }
  //********************
  //*  PUBLIC METHODS
  //********************
  //* Remove plugin.
  destroy() {
    this.ui.telInput && (this.options.allowDropdown && this._closeDropdown(!0), this.abortController.abort(), this.abortController = null, this.ui.destroy(), C.instances instanceof Map ? C.instances.delete(this.id) : delete C.instances[this.id]);
  }
  // check if the instance is still valid (not destroyed/unmounted)
  isActive() {
    var t;
    return !!((t = this.ui) != null && t.telInput);
  }
  //* Get the extension from the current number.
  getExtension() {
    return C.utils && this.ui.telInput ? C.utils.getExtension(
      this._getFullNumber(),
      this.selectedCountryData.iso2
    ) : "";
  }
  //* Format the number to the given format.
  getNumber(t) {
    if (C.utils && this.ui.telInput) {
      const { iso2: n } = this.selectedCountryData, i = this._getFullNumber(), s = C.utils.formatNumber(
        i,
        n,
        t
      ), r = this.ui.telInput.value;
      return this.numerals.denormalise(s, r);
    }
    return "";
  }
  //* Get the type of the entered number e.g. landline/mobile.
  getNumberType() {
    return C.utils && this.ui.telInput ? C.utils.getNumberType(
      this._getFullNumber(),
      this.selectedCountryData.iso2
    ) : qe.UNKNOWN_NUMBER_TYPE;
  }
  //* Get the country data for the currently selected country.
  getSelectedCountryData() {
    return this.selectedCountryData;
  }
  //* Get the validation error.
  getValidationError() {
    if (C.utils && this.ui.telInput) {
      const { iso2: t } = this.selectedCountryData;
      return C.utils.getValidationError(this._getFullNumber(), t);
    }
    return qe.UNKNOWN_VALIDATION_ERROR;
  }
  //* Validate the input val using number length only
  isValidNumber() {
    const { dialCode: t, iso2: n } = this.selectedCountryData;
    if (t === ue.DIAL_CODE && C.utils && this.ui.telInput) {
      const i = this._getFullNumber(), s = C.utils.getCoreNumber(i, n);
      if (s && s[0] === ue.MOBILE_PREFIX && s.length !== ue.MOBILE_CORE_LENGTH)
        return !1;
    }
    return this._validateNumber(!1);
  }
  //* Validate the input val with precise validation
  isValidNumberPrecise() {
    return this._validateNumber(!0);
  }
  _utilsIsPossibleNumber(t) {
    return C.utils ? C.utils.isPossibleNumber(
      t,
      this.selectedCountryData.iso2,
      this.options.allowedNumberTypes
    ) : null;
  }
  //* Shared internal validation logic to handle alpha character extension rules.
  _validateNumber(t) {
    if (!C.utils || !this.ui.telInput)
      return null;
    const { allowNumberExtensions: n, allowPhonewords: i } = this.options, s = (l) => t ? this._utilsIsValidNumber(l) : this._utilsIsPossibleNumber(l), r = this._getFullNumber();
    if (!this.selectedCountryData.iso2 && !en(r) || !s(r))
      return !1;
    if (r.search(ct.ALPHA_UNICODE) > -1) {
      const l = this.selectedCountryData.iso2;
      return C.utils.getExtension(r, l) ? n : i;
    }
    return !0;
  }
  _utilsIsValidNumber(t) {
    return C.utils ? C.utils.isValidNumber(
      t,
      this.selectedCountryData.iso2,
      this.options.allowedNumberTypes
    ) : null;
  }
  //* Update the selected country, and update the input val accordingly.
  setCountry(t) {
    if (!this.ui.telInput)
      return;
    const n = t == null ? void 0 : t.toLowerCase();
    if (!de(n))
      throw new Error(`Invalid country code: '${n}'`);
    const i = this.selectedCountryData.iso2;
    if (t && n !== i || !t && i) {
      if (this._setCountry(n), this._updateDialCode(this.selectedCountryData.dialCode), this.options.formatOnDisplay) {
        const r = this._getTelInputValue();
        this._updateValFromNumber(r);
      }
      this._triggerCountryChange();
    }
  }
  //* Set the input value and update the country.
  setNumber(t) {
    if (!this.ui.telInput)
      return;
    const n = this.numerals.normalise(t), i = this._updateCountryFromNumber(n);
    this._updateValFromNumber(n), i && this._triggerCountryChange(), this._trigger(Kt.INPUT, { isSetNumber: !0 });
  }
  //* Set the placeholder number typ
  setPlaceholderNumberType(t) {
    this.ui.telInput && (this.options.placeholderNumberType = t, this._updatePlaceholder());
  }
  setDisabled(t) {
    this.ui.telInput && (this.ui.telInput.disabled = t, t ? this.ui.selectedCountry.setAttribute("disabled", "true") : this.ui.selectedCountry.removeAttribute("disabled"));
  }
}
const cr = (e) => {
  if (!C.utils && !C.startedLoadingUtilsScript) {
    let t;
    if (typeof e == "function")
      try {
        t = Promise.resolve(e());
      } catch (n) {
        return Promise.reject(n);
      }
    else
      return Promise.reject(
        new TypeError(
          `The argument passed to attachUtils must be a function that returns a promise for the utilities module, not ${typeof e}`
        )
      );
    return C.startedLoadingUtilsScript = !0, t.then((n) => {
      const i = n == null ? void 0 : n.default;
      if (!i || typeof i != "object")
        throw new TypeError(
          "The loader function passed to attachUtils did not resolve to a module object with utils as its default export."
        );
      return C.utils = i, Ht("handleUtils"), !0;
    }).catch((n) => {
      throw Ht("handleUtilsFailure", n), n;
    });
  }
  return null;
}, Ht = (e, ...t) => {
  Object.values(C.instances).forEach((n) => {
    const i = n[e];
    typeof i == "function" && i.apply(n, t);
  });
}, C = Object.assign(
  (e, t) => {
    const n = new ur(e, t);
    return C.instances[n.id] = n, e.iti = n, n;
  },
  {
    defaults: Oe,
    //* Using a static var like this allows us to mock it in the tests.
    documentReady: () => document.readyState === "complete",
    //* Get the country data object.
    getCountryData: () => _t,
    //* A getter for the plugin instance.
    getInstance: (e) => {
      const t = e.dataset.intlTelInputId;
      return t ? C.instances[t] : null;
    },
    //* A map from instance ID to instance object.
    instances: {},
    attachUtils: cr,
    startedLoadingUtilsScript: !1,
    startedLoadingAutoCountry: !1,
    version: "26.7.6"
  }
);
var dr = /* @__PURE__ */ ps("<input/>");
function _r(e, t) {
  Si(t, !0);
  let n = Ut(t, "disabled", 3, !1), i = Ut(t, "inputProps", 19, () => ({})), s = Ut(t, "options", 19, () => ({})), r = Ut(t, "usePreciseValidation", 3, !1), o = Ut(t, "value", 3, ""), a = /* @__PURE__ */ $(void 0), l = /* @__PURE__ */ $(void 0), u = /* @__PURE__ */ $(null), c = /* @__PURE__ */ $(!1);
  const d = () => E(l) ? r() ? E(l).isValidNumberPrecise() : E(l).isValidNumber() : null, f = () => {
    var g, L;
    const h = d();
    E(u) !== h && (V(u, h, !0), (g = t.onChangeValidity) == null || g.call(t, !!h), (L = t.onChangeErrorCode) == null || L.call(t, h ? null : E(l).getValidationError()));
  }, p = () => {
    var h, g;
    (g = t.onChangeNumber) == null || g.call(t, ((h = E(l)) == null ? void 0 : h.getNumber()) ?? ""), f();
  }, _ = () => {
    var h, g;
    (g = t.onChangeCountry) == null || g.call(t, ((h = E(l)) == null ? void 0 : h.getSelectedCountryData().iso2) ?? ""), p();
  };
  Zn(() => {
    E(a) && (V(l, C(E(a), s()), !0), E(a).addEventListener("countrychange", _), o() && E(l).setNumber(o()), n() && E(l).setDisabled(n()), V(u, d(), !0), V(c, !0));
  }), Jn(() => {
    var h;
    E(a) && E(a).removeEventListener("countrychange", _), (h = E(l)) == null || h.destroy();
  }), Ji(() => {
    E(c) && E(l) && E(l).setDisabled(n());
  });
  function m() {
    return E(l);
  }
  function v() {
    return E(a);
  }
  var y = { getInstance: m, getInput: v }, b = dr();
  return Ps(b, () => ({ type: "tel", oninput: p, ...i() }), void 0, void 0, void 0, void 0, !0), Os(b, (h) => V(a, h), () => E(a)), _s(e, b), Pi(y);
}
export {
  _r as default
};
